In [42]:
import torch
from torch_geometric.data import Data
from torch_geometric.datasets import TUDataset
from torch_geometric.data import DataLoader
from torch_geometric.datasets import Planetoid
from torch_scatter import scatter_mean
from torch_geometric.nn import MessagePassing
from torch_geometric.utils import add_self_loops,degree
from torch_geometric.nn import knn_graph

Creation



<img src='data:img/jpg;base64,iVBORw0KGgoAAAANSUhEUgAABEIAAAL7CAIAAAB2tbepAAAACXBIWXMAAA7EAAAOxAGVKw4bAAAA
EXRFWHRTb2Z0d2FyZQBTbmlwYXN0ZV0Xzt0AACAASURBVHic7N19nMz1/v/x58xnZ/bS2nW5iFxE
JEqplPNTDilHHXW6UDgVKbkOKVchKqISq5xD8dWFogtR6tC1U3Fo44Rs5CKFtWvXslezM/uZ+f0x
Zs+6Zq35zOw+7re5uc3OfObzes12a2ef+/6832+bz+cTAAAAAIQPu9UNAAAAAMDZIcYAAAAACDPE
GAAAAABhhhgDAAAAIMwQYwAAAACEGWIMAAAAgDBDjAEAAAAQZogxAAAAAMIMMQYAAABAmCHGAAAA
AAgzxBgAAAAAYYYYAwAAACDMEGMAAAAAhBliDAAAAIAwQ4wBAAAAEGaIMQAAAADCDDEGAAAAQJgh
xgAAAAAIM8QYAAAAAGGGGAMAAAAgzBBjAAAAAIQZYgwAAACAMEOMAQAAABBmiDEAAAAAwgwxBgAA
AECYIcYAAAAACDPEGAAAAABhhhgDAAAAIMwQYwAAAACEGWIMAAAAgDBDjAEAAAAQZogxAAAAAMIM
MQYAAABAmCHGAAAAAAgzxBgAAAAAYYYYAwAAACDMEGMAAAAAhBliDAAAAIAwQ4wBAAAAEGaIMQAA
AADCDDEGAAAAQJghxgAAAAAIM8QYAAAAAGGGGAMAAAAgzBBjAAAAAIQZYgwAAACAMEOMAQAAABBm
iDEAAAAAwgwxBgAAAECYIcYAAAAACDPEGAAAAABhhhgDAAAAIMwQYwAAAACEGWIMAAAAgDBDjAEA
AAAQZogxAAAAAMIMMQYAAABAmCHGAAAAAAgzxBgAAAAAYYYYAwAAACDMEGMAAAAAhBliDAAAAIAw
Q4wBAAAAEGYirG4AABAqrrnmGo/H43K5TNO02+1Op9PhcCxcuLBJkyaW9JOdnZ2QkHD84zt27Lj9
9tsl2e12r9frfzAxMfHrr7/23//999+LDzZN0zAM/7/+R+rWrXte2wYABAExBgBwRGpqqsvlMgzD
/xt/VFSU2+3Ozs4uPuDGG280TdM0zWNeWOh2Rzqd8fHxH3300bm3MWnSpKVLl27bts3lckmqWbNm
ixYtZs6c2ahRI/8B6enp/qBimqbH43G73dHR0XFxcf5n1/3ww5/atjVN0x+BPB6PYRiHDx+Ojo72
H/DAAw8kJyefe58AAAsRYwAA/2MYRlRU1MGDByXl5ubGxcV5PB7/Uxs2bFi7dq3L5XK73Sd8bY0a
Nc69gauuumrz5s1RUVFer9fpdEZFReXm5n777bctWrSYPHnykCFDJLndbn+HTqczNjbW37PT6fSf
ITcnx+12O53OzMxMSdWqVXO73aZp5ubm+g8oLCw89z4BANZibgwA4Aj/xVcHDx7s2LGjz+fz+Xw5
OTlt27b1P7t23Tr/MZISExOrVasWHx8fFxcXFxfndDrj4+MrV658jg1cc801mzdvNgzD4XD85S9/
GTt27MMPPxwVFZWXl+dwOIYPH56SkiKpXbt2/vYyMzMbNWqUm5ubl5dXHLfat2/v8/kKCwt9Pt8P
P/zgcrlcLteUKVN8AXPmzDnHPgEAlmM0BgBwhN1ulxQdHV2rVq3jn12zerXX63U4HPfee+9rr71W
5tXnz5+/du3aqlWrZmdnjxo1avTo0f7HBw0adN111x0+fDg+Pv6VV14pWTouLi4pKSk6Oto0zYiI
E3yiXXTRRf5ZMfXq1SvzhgEAFmI0BgBwhM1mO8WzKSkpbrfb4XD07NnzfFSfO3eu/8KwVq1aFWcY
SUlJSSNGjKhSpUr16tV37tx5tqf1T+bx+Xxl2SsAwGqMxgAAjmIYhn9Y5hiHDx82TTM2Nrb1FVeU
edENGzZs3bo1NjbWZrMtWrTomGf79evXr1+/czl/8YJmAIDygdEYAMBRTphhdu3alZ+fb5qm2+2u
FJgDs3HjxqVLlxYdt3BZKaxYscLj8Xg8ntq1azds2PDcT1jshEurAQDCHTEGAHB633//fXp6elxc
XLVq1VauXHnffffVrl37pptu6tWrV90LLujRo8ehQ4fO5fy//PKLw+HIzc3171Hz7bffduzY8dJL
L61du3azZs26dOmyefPmMnorAIDygIvKAACnt2HDBklut3v37t29e/c+dOiQf2QmPj6+sLBw0aJF
mzZt+uc//9mmTZvSnT8zM9M/Myc9PX3EiBGvvfbawYMH/auiZWdn//HHH507d3744YfHjh17zAu9
Xq9/vOWEl40xJQYAyitiDADg9NLS0iT5t3NJSEj429/+1rJly02bNn3++ec7duyIj49PTU3t16/f
+vXrS3f+HTt2+C9m+/nnn/2R6corr7zmmmv+85///P777y6Xa+/evc8++2yjRo3uvffeMnxfAIAw
RYwBAJxeZmZmvXr1DMNo0aLF0qVLSz718MMPz5s3zzTNn3/+edq0aSNGjCjF+X0+n8fjiYuLO3Dg
QK1atd59993i/Wo+/vjju+++2zAMwzBeeeWVs4sxXq9/SOeEE34AAOGLH+sAgNNbvnz5b7/9tmPH
jmMyjKQ5c+bUr18/OjraMIwFCxaU7vz+PS6joqIkDRs2rDjDSLrlllvuu+8+wzAiIyN//fXX/6xe
XfKFPp/vFBeVeSWn0+l0OouKikrXGAAgNBFjAADnqmXLlv47OTk5pTtDjRo1JJmmmZiY+Nhjjx3z
7PDhwx0OR2Zmpsfj2bZ9e8mnTr2SstPpTExMrF69euXA6moAgPKBi8oAAOcqOjraf8flcpXuDHXq
1ImKinK5XDVr1jz+2QYNGyYkJHg8nvz8/K1bt575aePi4r799tvY2NiqVauWrjEAQGgixgAATiMl
JeX77783DCM+Pr5nz57HH5Cbm2uapmEYDoejdCUuuuiivLw8wzDy8/OPf3b3b79lZWUZhmGaZp06
dc7qzPXq1StdSwCAUMZFZQCA01ixYsXgwYOHDh3ar1+/ZcuWHfPsnj17/GuLmaZ5Qd26pStx//33
JyQkmKaZn5//yiuvHPPsW2+9ZZpmTExMlSpVrrjyypJPMXcfAComfvoDAE7jzjvvrFGjRkJCQm5u
7mOPPZaVlVXy2Xvuuce/HLOkl6ZPL378z3/+8zXXXHPFFVfceOONpy2RlJR09dVX+/eiGTdu3PYS
E2C++uqrV1991TCM9PT0Ro0aXdW6dckX2mw2/yJm5BkAqFC4qAwAcITdbjcCSj7epEmTW2655e23
35a0bdu2Dh063Hrrrc2bN9+2bds777yzbds2p9NpmmaPHj1Kbn+5cePGAwcOGIbRpEmTM6k+ZsyY
lJSU7Oxsj8fTpk2bBx988Iorrvj3v/+9ZMmSnJyc3NzcxMTEZ599tmzfMgAgTBFjAACn99prr6Wl
pa1atcput//6668zZszwz1SR5J+a36ZNm7lz555LibZt2z7xxBPjx493uVxRUVFz5szxeDx2u/3w
4cOS4uPjhwwZUnIh5mL+ThiNAYAKhR/6AICjnGy1seXLlw8ePLhmzZoxMTEej+fgwYOHDx8+fPhw
VFTUiBEjvvnmm2OOj4yMPNvSw4YNmzlzZqVKlTIzMyW53W6Px1OrVq0GDRq89dZb48ePP/4lERER
/uGjU6+8DAAoZxiNAQAc5Zgrykp65plnnnnmmS1btuzatevXX391uVwXXXTR7bfffsKD//vf/y5e
vHj48OGnOOHxevXq1atXr/Xr16empu7Zsye+cuVGDRt26NDhZMfbbDb/OAyjMQBQoRBjAABHMU3T
5/Od4oBmzZo1a9bstOepWrVq06ZNDcO45JJLzraHVq1atWrV6kyONE3T4/FIstlsZ1sFABC+iDEA
gKO43e6NGzd+/PHHcXFxRUXujh07lfpUjz/+uN1u/9vf/laG7fl99913kn777bdff/3V4XA4HI5T
Ry8AQDlj4+c+AMCvWrVqOTk5brfb/2VcXFxubu6PP/54hgMjx2jWrFlqaurAgQOTk5PLtE399NNP
V111lSTTNP3LDEhKSkrat29f2RYCAIQsRmMAAEc4HA6n0xkVFeX/0ul0Sir+8mwlJia+8cYbPXv2
LLP+SvB35fV6/THGMAyHw3E+CgEAQhMxBgBwxJNPPulyufxTTSQ5HA6Xy3Um02BO6Pvvvy+71o7S
smXLiRMn+u/7Fyiz2+3RMTHnqRwAIARxURkAAACAMMPylAAAAADCDDEGAAAAQJghxgAAAAAIM8QY
AAAAAGGGGAMAAAAgzBBjAAAAAIQZYgwAAACAMEOMAQAAABBmiDEAAAAAwgwxBgAAAECYIcYAAAAA
CDPEGAAAAABhhhgDAAAAIMwQYwAAAACEmQirGwAA4Cz8a8WKQperTZs2NWvWtLoXAIBlGI0BAISN
6dOn/+322++9996xY8da3QsAwEqMxgAAwsYbb7xhmqbb7S4sLLS6FwCAlYgxAIAwkJGR0bJlS4/H
43Q6DcPw+XxWdwQAsBIXlQEAQteOHTuWLFnStWvXhg0bZmRkZGZm2u32goICh8NhdWsAACsxGgMA
CFFFptmhQ4esrCyv1yvJMAxJhw8f9t8BAFRkjMYAAM5aq1atYmJibDZbw4YNj3928+bNtoARI0aU
uopZVOTxeDweT25ubm5urtvtNk1TgTwDAKjIGI0BAJy1m266aePGjYmJiRkZGWvWrGnTpk3JZxcs
WBAdHW2aZnx8/LksKRYZGfnKK6/k5eX5LyHbtGnT7Nmz09PTz7V7AED4I8YAAM7aQw899Nxzz+Xl
5bnd7kWLFh0TYz777LOoqKiDBw82bdq0cuXK51Lor3/9a/H9O++8c9GiRcQYAICIMQCAUmjUqFHL
li03b94s6euvvy75VHZ29v79+yVFR0cPGjTI/+CLL744Y8YM/3273W6z2Xw+n9frtdvtkmw2mySH
w+Hz+S6++OKPPvroZHW5nAwA4EeMAQCUxqOPPvrII48YhpGWlvbTTz+1bNnS//i8efPy8vIkNW7c
+O677/Y/mJ6evnv37pIvLw4k/ukukpxOp9vtrl279imK+gOPJP+kfwBAhUWMAQCURq9evSZMmJCR
kXHo0KF58+a99NJL/sefffZZj8dTUFBw2WWXFR988cUXt2/f3n/fPwJTnEO8Xq9/ExibzVZUVHT5
5ZeftrTb7S7b9wIACDvEGABAKbVv337x4sWSVq1a5X8kLS0tMzMzPj6+oKCgd+/exUf26tWrV69e
514xIuLIx5Y/CwEAKiw+BgAApdStW7cqVaoYhrF+/frU1FRJo0ePjo6OPnz4cHR09A033FDmFbmW
DADgR4wBAJRS586dnU6nw+GoVavWs88+K8k/6d8wjHvuucfq7gAA5RkXlQEASq9jx44LFiw4fPjw
J598Mm/evG3btklKSEiYNm1aycNWrly5dOlS//3i68H+Nz3G55Nkt9kcDkejRo2K1zc7XvFri+f6
AwAqJmIMAKD0hgwZsnTpUo/H4/F4hg4d6r+crFmzZlWrVi152JdffvnKK6+c9my1atVq2rTpKWKM
P72w7DIAgBgDACi95s2b169ff+PGjQosnex2u/v163fMYRdeeKF/CTKbzeYfT/FK/gXKFBiW8Xq9
pmnWr1//FOWY2Q8A8CPGAADOSZ/evR9+5BH/CInT6UxKSurevfsxx/Tr1+/4bFMKNpvNMAxGYwAA
/FkLAHBOHurbt1atWv77brf7TDZ+KTUjIiI2NtbpdJJkAKCCYzQGAHCuGjVqlJ6e7r//4IMPnr9C
V7W+ov6FF7rd7jZt2py/KgCA0GcrvjQZAIDSad269YYNGyTVqFFj7969VrcDACj/uKgMAHBOpkyZ
snHjxoSEBNM0b7nlFqvbAQBUCIzGAABK489//vMff/zhdDp37NhRUFAgqVq1ahkZGVb3BQCoEJgb
AwAojby8vAMHDuTl5Zmm6XQ6Y2Njn3jiCaubAgBUFMQYAEBpVKtWLS0tLSEhoXr16vXr13/ooYc6
duxodVMAgIqCi8oAAAAAhBmm+AMAAAAIM8QYAAAAAGGGGAMAAAAgzBBjAAAAAIQZYgwAAACAMEOM
AQAAABBmiDEAAAAAwgwxBgAAAECYIcYAAAAACDPEGAAAAABhhhgDAAAAIMwQYwAAAACEGWIMAAAA
gDBDjAEAAAAQZogxAAAAAMIMMQYAAABAmCHGAAAAAAgzxBgAAAAAYYYYAwAAACDMEGMAAAAAhBli
DAAAAIAwQ4wBAAAAEGaIMQAAAADCDDEGAAAAQJghxgAAAAAIM8QYAAAAAGEmwuoGAADlnFsqlA5L
OVKu5JK8kltyB/6W5pMipAgpToqV4qRoKUFyWNw4ACB0EWMAAGUsR/pVypEOS3lSkeSVigJ3TEmB
f49nl2ySQ4qSYqQ4qbJUXWoUvPYBAGHA5vP5rO4BABD2CqQtUoZ0UCqUoiRJplQoeSRJvqOjy/Ex
xpAk2QNDNBGSTbJJpnRYipLipQuly6RK5/etAADCADEGAFB6adJWKVtyS3mSW/J/qDgkr+SVfIFH
VOKO9+iTHDNN0wgc7Amcp0BySpGSpEipilRLukiKPy/vCQAQBogxAIDS+K/0i5QmSXJI9kA48QVG
UYyjM4w/qxQfU/yv7UQn9ycZM/CsrcTxZmBgJ0aqKjWULivLtwUACA/EGADA2dko7ZX2myo0VFRi
LKUMY0yx42OMJzDTxi3FSwmSzVRDQ9eUwTsDAIQNYgwA4IwUSCnSfp8KJa9NpuQ5boqL/0Iye2A4
RaeMMWfCdvS//hUCvIExGUNySDYpRoqX4qXWzJwBgIqBGAMAOA2XtFb6QyrwyWaTLZBDio4LJMWT
XsoqxpR8bfHLi5c7swUSjlMypCifIqU6NjWTKp9lCQBAeCHGAABOZZP0k3RQ8gbWH5NklAgSfv7Z
/MfHGG/gMHuJw85W8eYARYGzFQeY4jbskk9ySIZkkxpIfzr7QgCAcEGMAQCc2E7pv1JmidEPf3ox
ShzjO10sKX7WOPleMad1/B5n/tN6AzGp5OCPPy/FSTFSfRYAAIByihgDADiBT6Q/JE9gP8oTMk4U
Y0pOx1eJ0ZjjHf9gyVOVjD0nizHmcWs3GyVWDoiUoqVaUruT9A8ACF/EGADAUdKkH6T9kieQUkqm
iJKxwVciNhQrwxhzsiO9Ja4oO1mMMSW75JDipEipmdT0RD0AAMLU8X/hAgBUXOukX6V8yX26pZCL
HXPYGf5trGRK8Z70qBOzH3fn+OrFI0UFklvaJOVJV55lIQBAyGI0BgBwxKdSmuSSIiSH5A5MqT8+
Lfj5jk4jxQ+WdLLRmBPGmDMcjSn5wuMjkK3Eis9GicWaHVId6f+d5FQAgPBCjAEAyCMtlgqkfElS
tBQVuK/AjpY6LqKcIt6c8H5JJVPKaaf+n+zKgRPGmOIAYwSWLyvupIoUIzWXkk5XEQAQ4ogxAFDR
pUnLS4x4HB8qSrdKslWOHyBSYDqNIUVKhtRCujTYfQEAyhJzYwCgQvtD+lGKDEzoL0PnssLyuTBP
UtqUbJJHckhbpAgm/QNAOCPGAEDFtVVaJ/mkQrcMZxnHGEsyzMlKm5Ip+QJ3YqV1btmcutiC7gAA
ZYAYAwAV1F5pnVQg2aQIZylPUnwp2sm2fAkF/iXLiqQiKULySbmSnPpa8jEmAwDh6WTzMwEA5Vme
9JlklyIlUzooFZTqPL4T7YAZavwrAfj38fTfLwoss5YirbO6PQBAKTAaAwAV0adSXomVvoq3jDxG
6QZVQmooRiUm/ReHGf/gTKGUJxlSFamRlQ0CAM4aK5UBQIXzjpQuOY57/IS7u4RaJim14jfiTzKm
VChJMqRqUmepimWtAQDOGheVAUDF8qVUJEWd6Cnf0bdyxgjcbIEvIwKPFEn/tbg7AMDZIcYAQAWy
RsqQPD5++suQoiWHFCGZUrb0H6tbAgCcOT7IAKCiSJV+LtIhyXf81WPnQVCKlF5xjJFkSm5pl7TM
2p4AAGeMGAMAFUKOtFWyRUhSoeS2up9QEBWY/W9KHilX2if9anFTAIAzQowBgAphlXQosP+jn+10
N53b/P4Qn11jSnkl3qBbskmVpJ1WNgUAOFPEGAAo/9ZJe6T8wNpcKrEGcYVlSgXHxRi7dFBaZWVf
AIAzQowBgHLOI/0m+QJbxOBkijeTSfNou9XNAABOjRgDAOXcSmbCnDG35JKKHPrBZXUrAIBTIsYA
QHmWKh2UChmKOQNmYE/MfCk3Uqut7gcAcArEGAAoz76XXIzGnBmvZEo+KV9y2ViyDABCGjEGAMqt
bySb5JW8kv3oW4gvI2YJ//fEJkVINqlQet/ijgAAJ0WMAYBya695ZHiB0HLmitdwc0tZ0kErewEA
nBQxBgDKp58lGUcWFGZ55TNkSBElvl1eaYOV7QAATirC6gYAAOfF71KRVCQZksEU/7NhD/yRzy3t
tbgXAMCJMRoDAOXQTilbKuRysnPjk1xSitVtAACOR4wBgHJou1QgeSQFkozv6BtOyDz65t8z9Der
uwIAHI8YAwDlTZ70h1RQ4hEzsF5Z8U2STbJZ02A4KZRyGZABgNBDjAGA8uZnHZnZL4LKufFPkjGl
ney8AwAhhhgDAOVNWomLxvw/5bmKrHSMwMY7HtZ6A4AQQ4wBgHIlTTro/d+6ZD4GZM6NTXJLNkO/
WN0JAKAkYgwAlCs7JZ/xv4vKSrIf/UOfuf6nVTwGUyDtsLIRAMCxiDEAUK7kn2gGv02yB3KLnR/9
Z69IypJyrW4DAFCMzzIAKD/2SQcDYwjFK5L5/y0eeGEQ5sz5B7Vskkc6LP1ocTsAgP8hxgBA+bFZ
cut/E2NQVookp/SH1W0AAIoRYwCg/DgkFXLZ2HlgD4zJAABCBJ90AFB+HJKKJIMBmbIW4V+vjG8s
AIQMYgwAlBNZklsypdDfqtEMzzzglX63ugcAgB8xBgDKiV1SrOQ8eq+Y4gvMbEffzpN8yS4lSolS
vBQlqUQ//lUH/CsQREiVpBgpQnJIUSWOCVmmlGp1DwAAvwirGwAAlI0syS4ZUpGkwKVltuCuSxYp
Ze3X/r2KlEwp167qlylK8gbasAd6izD1xyZ5JJ8ht1duuxo0l2E7Mv8kNP/G5pWyre4BAOBHjAGA
cqIg8Nu/hespXyh9/MWeRbMfiizK9JmJB30XPTxz1tXXKluKKLGbpFOyZ2rptOStP633aP9hh7te
iy5DX3g0ruqRnW1CFmtVA0CIIMYAQDlxKPBLdvHOMEEeivFXbFSlTvbeAmVut7ujvDq0/ev/XnPt
ZQ7JI9kkh+SUKkmbN2vr6g8zd2xxRNulmAYdmiVWlivkP5ZYrAwAQkQo/80LAHAW/Hs1+nOLcaoD
z6NDbjVoqUuaXmczow257MrYnfJ50UEVeo+0558G4ypS6k+7stL3SpmegjQjMb5V2xucEZLkPJ9T
d84dMQYAQgQxBgDKieIf6L5ApAm+fKcKK6vOpR18jkSP0+uN2p/x63+2b1RciU8bh5SdofXrf/QV
mKqcqKq16jRuWufyyEKpSIoIzPApvoUUt5RldQ8AABFjAKB8yJdMqUgqknzW/fbvlJyxatzmz4qr
qkM+e56RtWlj2q+/RksOyQysB527Q7//5z0ZB6U8e37lS1rdX/NC5UhGiQwWLUVLhnWR7IRipAyr
ewAAiBgDAOVDfmj8uu+PUk2vVo2kBlJUlOw2M2fjum/y3EeuFvNK7iLt+umPnIydhplvdxsxDZq1
u+VGf/P+A8zABjjFd0IK15UBQCggxgBAeVBQYlq/VXNLbJIpeaSadXTxVe1UNTEiKqJIBbtTf8zf
K4ckySl5srRqxbu2wrwIU15FVa93Sf0rVShFSDapKJCFzMDNE0qbyYRapgKACosYAwDlwT4pIgR+
3Y+QTClban97TzniDh92yTi0f8vqrd/viZBsPhlS+s/ak/q9Lzc7Ug65o5td9eeiBBUGroUzpQgp
QoqVCvKO7I8ZOp9VPinX6h4AAAqljwYAQHjzBQZkCqQGLSMaN708MTFGpuk7mL12+fueAkXbpAKl
/rDWlf1bpAqkyMoX1Gt0+XV5UpFkBEZjHFJlKcLURbGKlaqE3kR/AIDliDEAgDLj9c9+kS6ooUta
dyzwRkqSWfTbf9elb5fhk7dQ6dtXO9xZkfKaiqpco0nShQ7/ZWP+i+KcUlGBYl2a/fg3N1/33Luv
5sRZ+o4AAKGJGAMAKDP+AZloKUeqfelNkQkNpSqR0oHUX3av2Rtj094D+u93i72uA7ZoI69yQtKV
XS68RJGB7WIMKclULad2rHb9uvQJrZ6b8/vabKvfFAAgBBFjAABlpngGvEdqdV3lyKoNZcSZpmS4
ftv8rS1fG7/bszdjn2FEHCpwKb5G+7/eFhktm+SUYqWiffrjOy1/fsXLj92bl/a7JLt8TgvfDwAg
VEVY3QAAoAw4JZ9kl8zAkIglvCV2rYmvqWrNr03f8r1X+TJz/9j6vXvf3b9t+FKewiK74mpeULPJ
5fUulSfwEp9bC2b9+O+FT1YyDxUd+N1RkCNFuFyFPp/stlBZIszORB0ACA2MxgBAeVBZskn2wM0q
/vkthn9zlXi1uvEuW3ysN8omIzdz19ov3/6/nes+secX5rmKcr2OBpf/KfoC5QVe67UrMjquelKd
2PiEpNq1YuIrKTouNjYu2qpMdiJ2KcbqHgAAIsYAQPkQbd0ITEn+FZ/tUpFkk1q1r+yo01Bej80o
Ksj45dO3X87Zt8WrIplGTKUq7W6+JdJ2ZBDJKxkR6vFYk398Oued/348cPq7vrjKKjiUl5dzMAQW
ki7m44MTAEIDP40BoDyIK7H9pc+6XRr9Hyo+ySPlSzFVdfHl19uiKiXYC8yCrIyM3w7/tknmIZmq
Wq9x1QsNd+A6NK+ULSlK3gSlGzJq1fVFxMiIiYmJjrTovZxQkRRtdQ8AABFjAKB8iJMcgWENy6eR
+JdddknOaLVq183njY2Wctwqys2wRZsyZY+Oa3L1n6OTVLzUsl1ySm4pSyqQsvJVaHPINE1Pkcfq
t1OSXUqyugcAgIgxAFBuRAYGqAyxIQAAIABJREFUZEJBkWRKHunSm6up7mV7XdH+bBPtVlWzirda
k+ZX3WhUUo7kknTiK+KOPBZSU+oNKcrqHgAAIsYAQHliD0z0t5Y/TfkXG4iL13Wt20Q5o2TKcEum
4VJi5ZqXXtC8aqF/SozFzZ6dkLrCDQAqMss/7AAAZcMRGL8IkZ/sPskn1YpWo0sud+Xa7c7q3ujK
+UaNPFW7vH23C5vJLcWG2+AGm9gAQIhg3xgAKCeipcOhsV5ZMVPaLzVsd/Wl9w87VOjKi4hwetU4
IrZFxzYuySYl+mTa/rfmcugLr7EjACjHiDEAUE40kv6Q3KHxq3bxiJBNatBKg14eZtqPrJscU6Si
OBVKEVKh7ch+nSV5JLepQzZTVSO9DqflKxYUM6QLre4BAOBHjAGAcqKR9K3VPRzPKxVIipUtkK8K
SzxbcKKX2KS4GmrQolN2dnalpJahs26BXaprdQ8AAD+bzxc6HxAAgHOySMqTvFKR1Z2ci0jJUyTX
LkVFyl1Ztnjrx5f8u/FESQ9Y3QkAwI/RGAAoP5xueZzKt7qNc1Qg2SMUe5EiJPvRozdWcUsOq3sA
AJQUIuvZAADKQFOnjBCb5V8KEZJPypUOSUVSdAjM9vE3UNXiLgAA/8NoDACEuuTk5MzMzEOHsgsK
8z2FbpfL5XYXut3ufFe+21Xg8XhuuKXzs6OfltRM2hj+MSZS8ki5kl2KkpwnmUITND4pUvIyMQYA
QglzYwAg1HXt2nXZsmVOp9PuNCTZTLNIbkletwxDjRpd9PPP24oPXiltlGIta7YM+Ic+/NN7ihcG
sHa9MlO6QOpqaQ8AgJKIMQAQ6nbs2HHZZZeZZvFv8kdiTOXYhAMHsr/8cmX79jcWH7xGWh/+AzIh
wifZJJsUK9WR2p/y4K+++/rf33wjrz3S4YyJiYmNjY2IiJDkcDji4+OrVKki6dprrw1K4wBQ/hFj
ACAM3HfffQsXLnQ6nZJM0/QZRZIipNtuu2vhwndKHpkjLZTsVg9flA/+D0hDcrr05yjVO93xTS5p
vHdPmsfldrvdkvz/vQzDMAzDbrcbhrFz587KlSuf564BoEJgij8AhIGrr746ISHBNE232+12u21m
hM2MaNSo2TEZRlIlqWYIzIkvN/xXtdU6gwwjaeHChVWrVHG73XFxcYZx5D+C2+3Ozc31eDx//etf
yTAAUFaIMQAQ0vbu3fv3v/993LhxmZmZpmlGR0c7nc7Y2Njq1asvXvzuCV9S8hdun2SQakrLK0mK
kKLP7PjWl1/1yCOPSIqMjPQPxUgyDMPpdFarVu3//u//zkuXAFAhEWMAIHTdddddderUefPNNx0O
R7t27apWrZqbmxsbG2sYxk033dSsWbMTvuoKKUpyBH4LR6l5JbsUIZ35GMqoJ0Y1b97c4/HExMRI
MgyjSpUqpmled911569PAKiAiDEAEIqmT5/esGHD5cuXV61addiwYfv37//mm2+mTJnidDrz8vJq
1qz52muvneLlNaQoSZI3MCATcfSN8Zkz4ZMcUpx02dm8atWqVQkJCcVf5ufnm6b5+eef33PPPbt3
7y7zJgGgYmKKPwCElvnz50+ZMmXXrl1ut7tr167vvvuuw/G/HeTbXHvtvr17P/vssyZNmpziJPnS
R1KhlB8YTzhmZMbHGgBnwCfFSNdJjc/yhfPmzXvsscfy8vIMwzBNs06dOm63+9ChQ3a7/YYbbnj/
gw8iDIIkAJwTRmMAIFRs3rz5mmuu6d27d0ZGRtu2bVevXv3hhx+WzDCSkpOTBw0adOoMIylGqiHF
ST7JK5mSN3DDmYuUEs4+w0jq3bt3586dTdM0TTMpKWnHjh1//PFHr169qlev/uWXXybVrDl27Niy
bxcAKhJGYwDAellZWUOGDPniiy/27dt36aWXvvDCC506dTrHc2ZLX0kHJK9kCwQYI7ClDKMxZ6Ky
dLF0RWlfXrt27by8vHfeeadz587FDz766KNvvfXWgQMHWrZs2adPn0GDBpVJqwBQ0RBjAMBi/fr3
f+/ddw8fPmwYxujRo8vw7/RLpAOSR1Jg8J0Yc1pGie9MvNT9HE61atWqF1988cMPPzz+qW7dun30
0UdRUVEXXnjhgw8+OHDgwHOoAwAVETEGACwzffr0OXPmpKam1qpVq1u3btOnTy/b8++XPpMOS0WS
f/VfYsxp2SUjcDHen6WLzluhjRs3Tp48efny5S6X68orrxwzZkyXLl3OWzUAKG+IMQBggffee2/c
uHFpaWkHDx7s1avX7NmzIyMjz0ehL6TtkosYcwaKd4mJkmySU+p2/ouuXbt2woQJn376aWJiYtOm
TWfMnHlV69bnvywAhD1iDAAE1bZt2/r27bthw4a8vLy2bdu+8sorTZs2PX/lcqVPpIOS/2c9MeYU
/DHGv8JyhNTqfA7FHGPJkiUvvPDCjz/+6Ha7u3TpMmfOnJo1awarOACEJWIMAARJZmbmmDFjli1b
lpOT06JFi5EjR/71r38NQt0N0iYpV5JkD0ySIcYcz/8NiZbipATpXNdYOHsLFixITk7+5ZdfKleu
fP311ycnJ1epUiXoXQBAeCDGAEAwjBo1atasWbm5ufHx8RMmTBg6dGgwq38s/RHIMMVrLrP48jH8
MSZGSpBut66NSZMmzZgxIzMzMz4+/v777585c6Z1vQBA6CLGAMD59eqrr06dOnXbtm2NGjW66667
Jk+eHMzqy5Z9GHvFbbUv0L8D4zBeyVdiCWYo8A0pkiTFSS2kKy3uSE899dQbb7yxffv2Jk2a9O3b
d9iwYVZ3BAChhRgDAOfLxx9//Pzzz6ekpJimedttt82YMaN69epBq/6vFcufmzh14y8/XdXhjk8X
vbpG2im5JTeXkx3HG/i3klRNCpH1wopMc/SoUfPmzSssLKxdu/bIkSN79epldVMAECrsVjcAAOXQ
jh07br311nvvvTclJaVNmzaffPLJwoULg5ZhduzY/te/3dr1r13/s35d8+YtBz9wh6Q2UpXAdWVO
yQhOK2HC/21xSBEhk2EkRRjG1KlTv/322w4dOmRnZz/66KOtW7desmSJ1X0BQEhgNAYAytKhQ4dG
jhr14ZIlhYWFTZs2Ddo8fr+MjIyRI0d++eWXBw4caNGixcCBA7t3/9/+jXnSMilfigwMy1Rk/g8/
m2RIhhQhRUttpLoW93Via9asGT169NatWwsLC6+77rrHH3+8bdu2VjcFAFYixgBAmenXv/8ny5fv
2bOnRo0agwcPHjlyZDCr9+jR47333nO73XFxcY899tj48eOPP+ag9KXklgqIMZKkCMkmRZlqYOgC
6UKLmzqN9957b8yYMVu3bq1WrdrFF1/8+uuvN2zY0OqmAMAaxBgAKAOvv/761KlTN2/eXK9evc6d
O7/00ktRUVFBq56cnDx9+vSdO3fWqVOne/fuU6dOPcXB26RVko0YIxVJ0VKsFONTgk3XW93SGZo3
b15ycvKGDRuSkpI6d+48Y8aMSpUqWd0UAAQbMQYAzsmnn346adKk7du35+TkdOrUad68ecHc62P2
7NnJyclbtmyRdO+99y5cuPBMXvW+tF9yBL40A0t1lW+2wAiMn3/nnBgpxqdqPnUMt7miU6ZMmTlz
5r59++rUqdO1a9eXX37Z6o4AIKiIMQBQStu3b7/rrrvWr18v6dJLL33hhRc6dQrelolff/11//79
t2zZEh8f3+ySS2bNmNH66qvP/OU/SLukIp/yJI9Nh6UYKeK8dWs5W4k1bfyxzZAipQipitTZytbO
yYMPPvjVV1/t3LkzKSnpkUceOeGVhABQLhFjAOCseb3eRx999M033/R4PPXr13/iiSd69uwZtOpb
t24dOXLkZ599lpub2759+wkTJrRr164U5/lO2ueRy6FcqVCKKL/LlxmSTXJIXsmU3FKRFCHFmWpg
6P9Z3d45crlcAwYMWLFihcvlqlq16qBBgwYOHGh1UwBw3hFjAODs9Ovf/6NlyzIyMgzDePDBB5OT
k4NWOiMjo2vXrtu2bXO5XJUrV544cWLv3r3P5YTLpSzJFfiyvO4nU7wWmaQiqUBySTHShdJfLG6t
zGRkZPTp02fZsmVOp7NOnTojRozo16+f1U0BwHkUbtcCA4B1FixY0KBBg3/Mnp2VldWnT5/8/Pxg
ZphHH330oosuWr16de3atceOHfvHH3+cY4aR1EWqHtgvxSd5AxtBlj/eElt/OiWHVK0cZRhJ1atX
X7p06bffftuuXbuMjIz+/fv/6U9/+vzzz63uCwDOF0ZjAOD0li9fPnny5NTU1Ozs7A4dOixZsiQm
JiZo1ceOHfvGG29kZ2c7HI7u3bvPnDmzbM//kfSbZARmwJe/v28ZkinlSXYpSoqUEqSu5fcium+/
/bZ379779u3Lzc3t0KHDU089xSYzAMofYgwAnMqePXvuuuuulJQUp9NZpUqVf/zjH507B29C+OzZ
sx9//HG32y3p5ptvfv+DDyKM8/K79ydShuSRPJJdMiRHYOwi3Pm/X6aULUVLCVI1qYvFTQXDq6++
Om7cuLy8vLy8vIsbN/7PunVxcXFWNwUAZYYYAwAn5vF4+vbt+9FHHx04cKBly5Zjxoy5++67g1Z9
1apVI0eOTElJiY+Pb9eu3csvv5yUlHReK26VvpcKJJ9USYrx6ZAtXPeWKV4/2idFSnbJF3iwhdTa
4u6Cas7cubNfeWXHjh2VKlW68847n3vuucjISKubAoAyQIwBgBMYPHjwv/71r4MHD7rd7mHDhgVz
Hduvvvrqqaee2rp16759+y6//PLZs2e3adMmOKV/ldb75LIdmSET1qMxxTFGkldySAlSS+liK5uy
zKhRo+bOnSspKiqKTWYAlA/EGAA4yquvvvrCCy+kpqbWqlXrjjvuCOYk/v379w8ePPiLL77Iz89v
2rTp+PHju3btGrTqxb6R9vjksskM2xhjBGKMPfAW4qQ7JKeVTVnvkUceWbp0aUZGRtWqVZ944olh
w4ZZ3REAlB4xBgCO+NeKFSMee2z37t1er/fGG2+cNWtW7dq1g1M6KytryJAh33zzzf79++vXrz9m
zJj77rsvOKVP6Adpq5QXnjHGPxnG5VakFOuUzac6Nl1vcVOhorCwcNCgQR988IHNZrvwwgsHDBjQ
q1cvq5sCgNIgxgCAtmzZ8vDDD69evTo+NvbChg3nz59/+eWXB636yJEjFy5caBhGfn7+Qw899PTT
Twet9ClskX6Q8qxuo3T8u8RESdEeXetQTav7CTU/bdjw6LBhKSkpLperZcuWw4YNu/fee61uCgDO
DjEGQEU3dOjQN954w+v1Nm/efODAgd26dQta6VdffTU5OXnv3r0Oh6PrbbclJyefp4XISm2TtF0q
kEzJLRUELtayWdqVTbIF5r2oxB1D8kp2yStFuHWVUy2taTA8fP3119OnT//yyy9zc3NvueWWiRMn
tmrVyuqmAOBMEWMAVFzDhw9fuHBhfn6+1+sdOnToxIkTg1b67bfffuGFF7Zt2+Z0Otu1azdnzpyq
VasGrfrZ2ij96laGXe4I2UMgydhKVC/+DPNvdxMpuaVI6VYpwYLWws+yZcvGjBmzadOm+vXrX3rp
pdOmTWvatKnVTQHA6RFjAFRE77///pNPPrlly5amTZvefPPNU6dOdTgcwSmdmpo6cODAf//737Gx
sddee+348eOvvvrq4JQ+R/+WNksOySsVSbIuyRgl7hRvCyMpSqoqNZPqWdNXGPvnP//56quv7t69
Oz8//7bbbps9+5W4uEpWNwUAp0KMAVCxfPHFF5MmTdq6dWtERMS11147a9as6tWrB6f0li1bJkyY
sHbt2qysrCuvvLJ///533nlncEqXlQxpg3RAckuuQHiwB70NUyqUHIEJME7J5tMFNjWRzu/eOuXd
jBkz/vGPf6SmpjZo0ODOO++cOnWq1R0BwEkRYwBUFBkZGXfdddfq1avdbnejRo3++c9/dujQIWjV
H3jggSVLlhiGERcXN3z48CFDhgStdJnLk1ZLmZJLypUkRZS4uCsIXNIhqbIUK8mnGjZdzyVkZeeh
hx5655133G632+0eP378hAkTrO4IAE6AGAOg/PN6vYMGDfrkk09yc3OrVKky/LHHHn7ooaBVHzdu
3Hvvvbdr1666det27949mDtpnm+bpM3SoUCGKauPk+PnvRzDkJxSdam2FLwV5SqSnJyc0aNHL168
ODMzs3nz5n379u3fv7/VTQHAUYgxAMq5kSNHvv322xkZGdWrV+/Xr9/IkSODVnrevHkzZsz46aef
EhMTb7vttnnz5gWtdDDlS79I+6UDUq5bEc4jm056JAUyiRH413e6i9CKP5O8kldySA7JJ/kkw1QV
Q7FSQ6nu+Xo3+J9du3aNHj36X//618GDB5s3bz5p0qTbb7/d6qYA4AhiDIBya9myZRMmTFi/fn3d
unU7duw4a9asmJiY4JRevnz5pKef3rxpU0xMzE033TR27NgmTZoEp7SFfNJ+aY+ULR2QXFKRZJMi
JJvkDaQRnXLcxpDsgahTJJmSXYqXqkqJYgFlC3z11VeTJk3asmVLYWHh5Zdf/tzUqVe1bm11UwBA
jAFQHn3++efTp0//9ttvJd18880vvvhinTp1glN6zZo1EydOXLt2bXZ2docOHaZMmVJh9+LIkfZK
B6VsyS0VSYWSKZmSL5Bqivkkm2SXDClWckp2KUlKlJKkIKVPnNy/VqyY/uKL69atMwyjQ4cO48eP
b9asmdVNAajQiDEAypU9e/Y88MADn3/+ebVq1WrXrj1lypTOnTsHp/Tvv//es2fPVatWSWrSpEly
cnKnTp2CUzq8ZEimVBRIMrbAJWfVpEiLW8NpzJo165lnnklLS5N0yy23LFy4sFIl1mUGYA1iDIDy
Y+zYsfPmzcvMzLzkkkseeeSRvn37Bq30gAEDFi9efODAgdatW/fp0yeYpYEgmzRp0ptvvrl9+/Z6
9ep16dIlOTnZ6o4AVETEGADlgX8ef1paWs2aNR944IGJEycGrfTw4cPfe++9rKysmJiYfv36sTot
KoiHH374448/zszMrF69erdu3V544QWrOwJQsRBjAIS39957b/jw4VlZWXa7/eabb37ttdfi4uKC
U3rWrFmTJk3Kz8+vVKlShw4dFsyfb4+ICE5pIBR4PJ7u3buvX79++/btiYmJEydOHDhwoNVNAago
iDEAwtV33303cuTIlJQUwzDatWs3ZcqUFi1aBKf0+++/P3ny5A0bNkjq2rXr3Llzq1SpEpzSQKjZ
u3fvE088sXLlyuzs7FZXXDH00Ue7detmdVMAyj9iDIDwk5mZedNNN6WkpDidTqfT+dprr919993B
Kb3uhx+63X33zp07JdWvX3/ZsmVBy05AKNu1a9ftt9/+66+/SrrgggsmjBvX7d57rW4KQHl26l3I
ACDk9Ovfv27duikpKS1btnzppZdycnKCk2H27t3bpUuXq6+6aufOne3atVu0aNHOnTvJMIBf/fr1
169f/+57711xxRV79+7t9eCDHTt2TE1NtbovAOUWozEAwsaYUaOWLF2anp7ucDj69+//5JNPBqdu
Tk5O//79V65c6fF4Kleu/OSTT/bu3Ts4pYFwtHjx4v79++fn58fGxrZt2/all16qX7++1U0BKG+I
MQDCwIIFC0aNGpWenl6jRo0mTZp8/fXXQSv9yCOPLF68WFLlypX/8pe/vPzyy0ErDYS15557Ljk5
OSMjo0qVKn/605/effddqzsCUK4QYwCEtO+++2706NHr1q0zTbNLly7PP/98w4YNg1P6+eefnzNn
zr59+2rWrNm1a9dp06bZ7VyIC5yd5557bu7cufv3769Zs2b37t2DuRg6gPKNGAMgRO3YsaNPnz7r
1q2TlJiYOHv27C5dugSn9Lx586ZMmZKVlZWTk3P3nXfOfe21qKio4JQGyqX77rtv0aJFsbGxMTEx
PXv2nDJlitUdAQh7xBgAoejxxx//4IMPsrOza9Wq1a9fv/79+wenbkpKypAhQ3788ccqVaq0atVq
4sSJrVq1Ck5poHzbv3//2LFj33///YMHDzZu3PjRRx8N2v/XAMolYgyA0JKcnPz888/v3r27Xr16
ffr0Cdo8/u+++27KlCkrV650Op1t27YdOXLkDTfcEJzSQMWRlpbWt2/fZcuW1ahRo3nz5v3797/z
zjutbgpAWCLGAAgVb7755pNPPrlr1664uLhbb7311TlzYuLiglA3JyfngQce+OCDDxITEyMjI6dM
mXL//fcHoS5QYS1fvvyFF17YvHlzZmZmmzZtZsyYceWVV1rdFIAwQ4wBYL0tW7YMHz48JSXFbrdf
f/31Tz75ZPPmzYNT+vHHH//www+3bdvWrFmz3r17P/bYY8GpC+DTTz995plnfvzxx9jY2A4dOowZ
M4aNmACcOWIMACvt379/6NChX331VVpa2jVt2jw9aVLHjh2DU3rSpEnz589PS0uLjY3t27fv008/
HZy6AEqaM3fuxKee2rNnT2Ji4o033jhnzpzKlStb3RSAMECMAWCZESNGLF261L8S6/Dhw/v27Ruc
uu+///6ECRM2bdpUtWrVTp06Pffcc3Xr1g1OaQAn9Pzzz8+bNy89Pd0wjPvuu2/atGlWdwQg1BFj
AFhg2rRpCxYsSEtLq1Sp0t///vegbSXxxRdfTJkyZe3atS6X67bbbps8eXLQdqEBcFpjx4794IMP
du3alZSU1KtXr6Ct8AEgHBFjAATV+++/P3r06N9//93hcHTs2PGNN96IiYkJQt1du3YMGDDoxx9/
TEtLu/rqq5999tkOHToEoS6As5KXk9Pj73//4osvcnNzq1WrNnr06KFDh1rdFIBQRIwBECQ//fTT
iBEjvv7669jY2BtvvHHUqFGXX355EOpmZWVNnjz5//7v/3w+30WNGw8cMKBnz55BqAug1Hbv3j1h
woR33nmnUqVKTZo0GTRo0N133211UwBCCzEGwHmXk5Nzzz33/Pjjj1lZWU2aNBk7dmy3bt2CU3rw
4MELFy4sLCyMjIwcOXIkC5EBYSQlJWXMmDGrV68+fPhw27Ztx40b16lTJ6ubAhAqiDEAzq9Ro0Yt
WLAgPT29adOm999//4gRI4JT96mnnlq0aNHBgwcdDsdtt902c+bM4NQFULaWLFny3NSpmzdtioqK
uuKKK+bNm1enTh2rmwJgPWIMgPNl1qxZc+fO/emnnxo0aNC1a9fp06cHp+7ixYsnT568YcOGunXr
drnllqnPPVepUqXglAZwnixcuHD69Onr16+PjY29/fbbp02bVr16daubAmAlYgyAsrdy5co+ffrs
37/f6XS2aNHik08+SUhICELdr776qkePHunp6fHx8QkJCR999FHQttEEEASPP/74P//5z6ioqPT0
9L///e+vv/661R0BsIzd6gYAlCtbtmy58cYbb7311gMHDnTs2PGzzz77/vvvg5Bhdu3a1bVr13vu
uSc9Pf3aa6+dP3/+jh07yDBAOTN16tRDhw716tWrbt26H3/8ccOGDZ966imrmwJgDUZjAJQNl8t1
zz33rFq1yuVyNWzYcNKkSbfffnsQ6u7atWvIkCFr1qzJzMysW7fus88+e++99wahLgBr3XPPPcuX
L5d0wQUX9OjRY+zYsVZ3BCCoiDEAysBDDz00f/580zQbN27co0eP8ePHB6duv/795732mmmatWvX
7tGjx+TJk4NTF0Ao2LVr1yOPPPLDDz9kZmbWqVNnxowZd9xxh9VNAQgSYgyAc/L000+/9dZbqamp
1apV69OnT9CCxKRJk2bNmpWZmRkdHT106NCJEycGpy6AULN169bBgwevX7/e5XK1bNly4sSJ7du3
t7opAOcdMQZAKc2fP3/EiBH5+flut7tLly5Lly4NTt1x48a98sormZmZkjp16rRixYrg1AUQylau
XHnXXXd5PJ6CgoJLL7309ddfb9WqldVNATiPmOIP4Kxt3rz5+uuv7927t8fjueGGG1avWROcDPPx
xx+3ufbaSZMmZWdnd+7c+Ye1a8kwAPw6dep06NChf/zjH82bN9+yZcsNN9xw11137d271+q+AJwv
jMYAOAu///77wIED16xZk5+fX7t27RkzZ958001BqLtmzZqxY8du2LAhMzPz+uuvnz59On9nBXAy
c+bOfX7atH379lWvXv2qq66aNWsWm8wA5Q8xBsCZGjBgwJtvvhkTE+NwOB544IHgTEfJzs7u3r37
p59+GhcXV61atQkTJtx///1BqAsg3I0dO/att97av3+/w+Ho2bPnyy+/bHVHAMoSMQbA6c2aNevl
l19OTU1t3Ljx3Xff/fTTTwehqNfrHTRo0Mcff7x79+5mzZo99thjvXv3DkJdAOXJyJEjX3/99Zyc
nNq1a/fp02fEiBFWdwSgbBBjAJzKwoULX3zxxV9++cXtdnfq1GnJhx9GGEYQ6o4bN27evHkul8tm
s/Xo0eOll14KQlEA5VX37t3ffvttSUlJSQMGDGCTGaAcIMYAOLHt27cPGjRo3bp1pmm2b99+4sSJ
zZs3D0LdN998c8yYMXv27KlTp06HDh1mzpwZFxcXhLoAyrdt27ZNnDhx5cqV+fn5F1988Ysvvtiu
XTurmwJQesQYAMfKzMwcOXLkN998s23btiuvvPLJJ5/s2rVrEOouXbp08pQpO3fsSE9P7969+8yZ
M6tWrRqEugAqjpSUlDFjxqxYsaJGjRqtWrUaMWJEhw4drG4KQGkQYwAcZcKECS+//PKBAwcaNGhw
//33jx8/PghFt2zZMmDAgK+++iopKemiiy566aWXrrzyyiDUBVAxLVq06Pnnn9++fbtpmp06dZo2
bVr9+vWtbgrA2SHGADhi9uzZc+fO3b17t9Pp7NOnz7jx44MwDWbv3r1jx45dtmyZx+O5+uqrB/bv
3/X22893UQCQNH/+/JmtohY8AAAgAElEQVQzZ27cuLF27dodO3Z87rnnWJcZCCPEGABatGjR9Jde
+mHduujo6C5duiQnJwfhszw/P3/gwIEfffTRgQMHkpKSxowZM3DgwPNdFACO8fTTT8+YMcM0TdM0
77jjjnnz5lndEYAzQowBKrTU1NQRI0asXLmyUqVKTZo0mTFz5lWtWweh7oQJExYsWJCVlZWQkNCv
X7+RI0cGoSgAnMzgwYNXrFiRlpZWs2bNAQMGDBkyxOqOAJwGMQaooLKzsydMmLB48eLs7OzGjRuP
GzfujjvuCELdF1988f+zd+fxTdT548dfySRp0qR3gXJfFkFuRQQRxQVRFgUPlBV18UK/HnihP3XF
c10WL0RBXATPVfGARVi8WPFAEORSQARhuQUKpQe9kiaZ5PfHdLJpy1HaJJO27+cjjz5CMp15R9OZ
eX+O9+fDDz/cuHFjZmbmpZde+vLLL8fgoEIIcUJ+Vb1/woR33nmntLS0Xbt2DzzwwM0332x0UEKI
Y5I0RojG6LHHHps+fXpiYmIgELj//vvvu+++GBx03rx5jz766ObNm10u15lnnjlz5szs7OwYHFcI
IWpu//79991338KFCxVFyczMnD59+vDhw40OSghxFJLGCNG4vPnmm6+88sq2bdvS09MvvfTSF198
MQYHXbt27eTJk7/99tvS0tJBgwY99thj/fr1i8FxhRCidjZt2vT//t//W7duHdCzZ8+HHnpo0KBB
RgclhKhE0hghGoslS5bceuutR44cOXz48FVXXTVz5szU1NRoH3THjh133nnnTz/9FAgEmjZt+tRT
T10mhciEEPXEp59++vjjj69duxYYOnTo1KlTu3TpYnRQQogKksYI0fDt379/7Nixq1atslqt3bp1
mzx5cmw6Q8aPHz9nzhyfz5ednX3NNdfce++9MTioEEJE1syZM6dNm5aTk6Moyvnnnz979myXy2V0
UEIISWOEaNB8Pt9dd9/9r3nzDh06lJ2d/eSTT1599dUxOO5jjz32wQcfbNu2rWnTpvfee68UIhNC
1HcvvfTStGnTcnNz09PTL7zwwilTpiQmJhodlBCNmqQxQjRYd91117Rp09LS0lRVveWWW5577rkY
HHTKlCkTJkwAbDbbsGHDPvnkkxgcVAghYuOaa66ZP3++2+3OyMi4dsyYqVJrUQjjSBojRAM0e/bs
l1566ZdffsnKyho1atS0adNicNBFCxc+8dRTa9euTUtLGzp06EsvvdSsWbMYHFcIIWLJ5/ONGzfu
888/Ly4uTk9Pf+KJJ6QusxCGkDRGiAZl4cKFjz766N69e4uKigYNGvSvuXOToz+P//PPP3/yqad2
7tjh8XhSUlIWL17cuXPnaB9UCCEMVFxcPGrUqMWLFzscjrZt295xxx133nmn0UEJ0biYjQ5ACBEZ
Bw8evOCCC0aOHLlhw4bTTjtt0aeffvXVV9HOYbZv3z5w4MA//vGPP65c2aJFi2nTpu3Zs0dyGCFE
g5eUlPTll1/+8ssvffv23b179/jx47t06bJq1Sqj4xKiEZHeGNEwlUE5FMIRcIMXSsECgbBtTGCH
BHBCJmSAzbB46+q666776quvcnNzs7Oz//rXv44aNSraRywpKbnrrru++OKLAwcOdOnS5Z57771l
3LhoH1QIIeLQsmXL7rrrrl27dgHdu3d/5pln6vXSWH44DPng0S+aR8Bf+QJqgySwghkyIRmSjAlW
NGqSxogGYg9sg9/BD2ZQwAQ+CIAV/FAOZjCDSf/pBxNoVTN9+k8/+KB5kCEmUoz8QDU1evToefPm
AcnJyTfddFNs5vFfc80177//vsPhUFV17Nixr732WgwOKoQQ8WzGjBmPP/641WrNy8vr1avXggUL
srKyjA6qRrywFnK8BBRMCgHwggpm/ZoYAHNYGmMFE3jBByZIAEW7wgYxl9PSTja0MPIDicZC0hhR
j/lgN+yHPCiDYvCAHZTKm5kqPzEf7a2g/lN7omU4NmgKraEN8ZjSfPjhh4899tjWrVtbtmw5cuTI
l6dOVazWaB90+vTpL7zwwuHDh1NSUq688soXX3wx2kcUQoh65LVZs16dMePnn3/OysoaMmTIzJkz
47Yu80bYAn4AVD0tCYIJlMpX0kDlX1QB/XKpvaU1Haph25j9NLfQGlpDRrQ+gWjsJI0R9U8+7IMc
yIWA3q/ihQD49D7ucFX+Gaw2tCz0uqny6wrY9LO5E7KgW3wMPJs3b96zzz67ceNGt9s9YsSIN954
IyMj6peJF1988a233tq/f//hw4dHXX75+x98YI1+1iSEEPXRpEmTZsyYsW/fvnbt2l00bNirM2YY
HVGFEvgZ9oEPVAjoVzqt10V7MagPajgqtfKTSqmLvjez3mMTBAvYIAsyoVMUP5lojCSNEfXJb7AV
/HAEPOAFBSwn+q3wDQJ6A5J25q1+mjaH5TPBsFO8dlpPhmRoB+3r/mFqZd++fX/60582bNhQVFQ0
ePDghx56aMiQIdE+6IoVKx544IHly5dnZmZ279592rRpXbt2jfZBhRCivrvt9tuXfPWV1n194403
PvroowYG8ztsggOQBzaw6CPBFH3iqF9PbLT2QdOxdxVqDQw90YYwWPWkJfSKtn8PmCARkqAZ9I/6
ZxWNhaQxon74EXZCgZ5mxF4o4dGOngI9IcYFucaPHz9//vycnJzevXvfd999V199dbSPuGHDhmee
eWbhwoVAz549n3zyycGDB0f7oEII0WCUlZXdc889c+bM8Xq9p5122t3jx19/440xjmE97INcvcsl
/BqqzRQNHPNX68R0jOt1EDJgIDSPznFF4yFpjIhr+2EreKEQ3OADb3yUCbdBArigPfSI/uEmTJjw
6quvut1ul8t17733PvXUU9E+ol9Vr7ryyu+//97r9drt9hdeeOHaa6+N9kGFEKJB2rdv33XXXffN
N9+4XK62bdv+5S9/GTNmTAyO+xNsVfGZwYQa1t8Sfud3nF6XKNGypgxoAT3icuqpqC8kjRHxaxGU
QnnlkWCBaudc9Si/GnWmsMkzTmgBUSquOWfOnKeeemrLli0tW7a8ZMSI2Aywfuyxx95+++3CwsI2
bdrceOON9957bwwOKoQQDdtXX301a9asTz75xGazDRo06NFHH+3bt2+UjrUedkARBKA8iN+LIwGM
TmMUfYR2qJCAE5pCH71qqBAnRdIYEY9Ww38hVy8dpk0WVI7WDxM0KI1BT6jM+iMJToNukdv/3Llz
p06dun79epfLddZZZ73//vsxKHczefLkN998c+vWrU2bNr355pv/9re/RfuIQgjRqCxatOiJJ574
+eefMzIy+vbtO3ny5MjONsyDtXAYiqAMTPo0GK3+jRo2rUUT+96Y0BpuPn1ogwOS4Q8gdWPESZE0
RsSXg7AqSImJYigHh778i5bAaF9WU1g+Y2AaE/rL0U7H2qUiHc6DJnXb8969e++8886FCxempaV1
7dr1oYceGj58eN12eWKff/75hAkTtm7d2qRJkyFDhkyZMqVJkzp+DiGEEEf3xhtvzJgxY+PGjYqi
XHbZZTNnznS56tohUQjfwBEIgg9KoBwACyTorYHERxpDWDE0rz7aLRn6wGkxj0fUX5LGiDiyAnaD
F8r1tbcSKicwIYanMYpeoSUUmDb4LRWSoTX0AMfJ79bn8/3lL3/58MMP9+7d27t37wcffHD06NGR
jPtoli5d+sQTT6xduzYQCFx22WUTJ07s1EmqYgohRNRNmTJl1qxZW7Zsad269ZVXXvnMs89alGMV
Oj6BjfArlOgVk7WZMKG5+ya9ArISlsbEPoHRWPRDa8GEJu1YIBGawpAa1CAVAkljRJw4ACt8FFvx
AJXPvJpQH3T1RWCM6o0J0a45/rBXtKUzzznJGvkTJkz47LPPtm3blpycfPfddz/++OORjPJoVq5c
+cADD6xdu9br9fbv3/+FF16I3kBtIYQQR3XXXXctWLBgz549bdq0GTVq1AsvvHCye/gGcsAddiUK
remsXTrVsIEMocvoUa+qMXCsFKUc/OCCBJXuCj1jGpSolySNEcbbCNuhyI3XUZGT1Pc0RpMKZ8Ip
NdjDggULJk2atHHjxqysrKFDhz7//PN1H11wfD6f77777nvrrbesVutpp512zz33jBo1KqpHFEII
cSz5+fkPP/jgt0uXbt26NTs7e+LEiX/+859r8ou/wE+APpAsdEGsksZoAvpkzmBYneX4SWOC+hwe
E6h+sixcGNO4RP0jaYww2OdwGMrAoneFhwv98zhnvbhNY2yQCi2Ou9TX4q8WT5ky9YdVK3zl/gsG
/+HZZ5/p3Cm6q9GUl5f/7W9/e+uttw4ePNikSZNHHnnktttui+oRhRBC1ER+fv748eO//PLL8vLy
U089dc2aNcfffg38DP7K5b9Cyzejv6I9D89nTGE/Y+9YF3RFH2YW0O8HWsIfajVCWzQSksYIw/hg
gd4JXqY3w4SvHBz+1ayPaUyC3rDUCs6t9m5x8ZHrrhu74PPP8Xq79erxzDPP/HHoRdEOddKkSTNn
zszPz09PT7/wwgtfe+21aB9RCCHESVm2bNkTTzzRqlWLt9565zibLYdtcET/Z2jGvHa5DIb9pPLz
KrNiqmQ+MXD8eS9aXR8tk0mFRBgEGbGJTNQ3ksYIY+yBleCGRPCBGwohIWzmH/U/jdGGhZUBkAlD
IUl/a+LEia+//npxWVnbdu0eeGDC9dfWaPBAXcyZM2fatGlr165NTU0dNmzYCy+8kJEh1wUhhKiX
lsJv4AVr5Qn9WoeGpb6lMaGJ/uiXVJM+mNwKCdALIlmUWjQUksYIA+yDr6BUH6SrnUC9Yc9DhVpC
KcqxSrcYnsMchxazqpeIccHZ0AqA7t17HinIuXzUqKlTX4l2GPPmzZs4ceLvv/8OnHHGGa+++mqX
Ll2ifVAhhBDR4IHFUAjFEABbter/5srLEgRimJ/UUPULunYpD+phax/Eolcw80EaZMOA2MUo6gdJ
Y0Ss7YaVcCSsFmT1hpnwc1w8JyrHEkpgqrzo9TPSQivYsDWnfXNHUlJKVMM4ePDg1VdfvXr16pKS
knPPPfeRRx4ZOnRoVI8ohBAiejzwNeSBJ2xpS8IaAauvrhY+JSbOb/gUfc3rUPmBAPjAB3ZIg05w
psExivgiaYyIqf3wDRSDEzzgA45WJqVBpjGARV/eq0OUA9DmiX7//fcHDx7s3LnzAw88cO2110b5
mEIIIaLoCCyFcn1dy9DSloSlMSH1NI0JJWOhsqUBvYZBCtjhVDjDyBhFfJH1hUTslMEyKAQ/OIyr
kRIDx0q93BCEnyABWkbt6OPGjVu4cGFhYWFycvJtt902derUqB1KCCFELBTAv8EXBBOqPlw5WG2i
S8NgCitUoE3d8YMH3PArpNRsJQPRGMS+XLhopArhCyjQz0pq2Jn3OKfg+tgVcxw2CEA+LIOcKOz/
ueee69Chw5tvvgncf//9ubm5ksMIIUQDsBjyoMxUsUakdnEMdcU0SFo/kgIJYAYflEIJrIfdBocm
4oX0xohY2A/fwkF9tJU17Cx8VA0se9Fo52K33qT0LYw6+b/Afv37f/fttwkJCVVenzx58uzZswsL
C/Py8kaMGLFgwYIIRS2EEMJgC+GwvsBlaPaLJnwMdviYsfp+GQ1U/qdZnzCjQiGsBjO0NiY0EUca
ag4v4ssy2AEOsOg37v7KJ9xjFSJrSFTwggkSwAZH4POT3MPEiRP/u23buHHjwl9csmRJz549H374
4YMHD55xxhnr1q2THEYIIRqMz6AAHJCkL0em6HU+Q/dwSoO7nzPrC3oGoBwCkAApYAUvFMC3/nqf
qom6a2BfexGPVoMHksGkn3+pPJCsyrcwzucg1pF2araAAkWwqsa/+Omnn86cObO8vPy7777bv38/
sHHjxosuumj06NHbtm3r37//p59++uWXX/bu3Tt6wQshhIiln8ALqn7hMIUtS3Cc5r+GdBlV9Z6l
UIeMNiNIsbDE4NCE8aRSmYiupbAPvOAN6yMOPTGH/Qw1q4R/IxvYzMUqQ8gUsEG7mtXCT0lJsVqt
ZWVlVqvVbrdnZGTk5OQUFRW1aNFi8uTJY8aMiUbAQgghjLIXNkIhlIINgFLg2OU96/tlVDl2tKEF
pkPrNKTo3VODYxGaiFMyN0ZE0e+wB4rBCuZjZC+NmQ88sBuyoelxtxw3bpzP5/P5fID2pLi4ODs7
+9prr33ggQdiE60QQoiYCcBvcATcoEangyV8qZk4ZwFV74qxgRnccAScsB9aGB2eMIqkMSKKftU7
YcrDqtrXo5ahGLBAOfwEFx57mwULFnz00UfVX587d252dnb0YhNCCGGUr6AAyvT1YeTSqa3p6dcn
zATBDX7YLGlMIyYN4iJavoEDENS/ZKGumKD+ONbkPFPYo2Ez651UB2DZMbcKjh9/h9Vqrf7GunXr
ohicEEIIg+yBQ1CmLxIN+PQZMtol9fiXyBpeRoPx1xUTOPaDyt1HWvBe2APfGBavMJikMSIqDsDv
+uIwCeDQx/USlsagT91T639pyFrTWpVMsE+l7GgbjPjj0KKC/OLiYlXn9Xrdbrfb7Z4xY0aswxVC
CBF9a6EE3GEtgF7wVLuzD2Up0b6MHie7qBJJXag1OJA57P7BDmbYDxtPpl6OaEgkjRFRscnoAOqd
AKys9uKsmVO++f5rt9cbSmC8Xq+qVlytNm3a9NqsWTGOUwghRFSthPxgRW1PUYVS7Z8OMMmCmI2V
zI0RkbcV8qqtXSVCApX/4yhaC5PCYdgHLcPe2rj5txEjrjSbFUWxJicnA2azuX379tq7WVlZp3To
EMPAhRBCRNcB2A5+E6b4G/EVP0JdT2awgwnK4Uc4y8ighAGk4LKIvI+gsHKKrIb9DH3hGvzUl+pC
S3+G15626yuXeaEpjDQsOiGEEAabBzlgAttxh4ppF1BzrFKdE7ZLKgYlXQpYwA9mcMIfIdmIMIRR
ZFCZiLBvoQxK9fmIauUcRoQLhP1n8UEpHIJfjYxICCGEYbbAISgB74m2jMPZ+UbR2gcD4IOfDY5F
xJqkMSKSSuAA+MByjDOsnHYBs74As1lvS9PWBNBelDRGCCEap5VgAitYZGB2jQX0n37YC3kGhyNi
StIYEUkboBz8eutIlUJkksNotL86rRS1VX+u6ouEeqRAghBCND4boAwCkKDPmQx11zfCMdg1F/qv
pC2IucbIWESsSRojIulA2AklWO0hQsL/a6iVh5YpsNeAiIQQQhhpqz6QIVA5hxHHUX3NhnwoNCwc
EWuSxoiIWQKloOpT7qT16DjMx3hosySPwBajIxRCCBEzh8GjF4CRO7OTEhruYdaL5cgaMo2H/LGI
iMkFv35CkWr3x3esNMYMKpRKh4wQQjQmP4NfvxcXdaFCvtExiJiRvxcRGTugHAjrEEc6ZGolAB7I
h8NGRyKEECI2CoIyiixiPDKiodGQ5S9FZGwBLwT11EUmJh6V/0QbmPXNymATnBf1iIQQQhjsS/Cb
jtL8F97SfNSLacwmnZ6wzTt+pr96QIXN0NnoSEQMSG+MiIxiQO8T15jCHuEU/SGOwy+9MUII0TgU
BHHLrJgI8YMXCqDE6EhEDMifjIiAnWH9DNXzFlELKpTAHqPDEEIIEVVHIGiq6EOInz6Nes0KKvxm
dBgiBiSNERGwGVQIgKkGX6nq5RFFiKovHqqVW5GlMIUQomH7L3iBykMYpEGw1mz6cI8dBgciYkHS
GBEBh6RMZISoYZVqVKlXJoQQDd0efVR2FdIzU2vawPVyWUu6EZDbTlFXB6AUfPo5t/qql1E9FysN
brKNCRQwQRDKwG10PEIIIaInTx/LIAtGR4p2P+CTgdmNgKQxoq7WgAswKJGwQCLYG+KCm9pHW210
GEIIIaLkF0iARENHkcWy5TFmLJDQgNo3xbFIwWVRV4V+fMZ9j7TaaGrYw2ZYLBFQ/Zz7uwfsBkQi
hBAi2rbVoBB/9GgZS6DyK4Gw6SXUz4msWsw+KDU4EBF1ksaIuvJbKAerQU04fvCFLbhJ2JPqXY1K
5TOySf+nEsftT2bJYYQQooHKBw9Y9ctQlZRGyyWil0iYwnZeH9OVY1EgAKVghhJ9wIhokGRQmaiT
7frkjZNKAyLYex7qhAmGDS/W9h+onN5olLDf8oeFEbcD0jyw0+gYhBBCRFwAyvWL1FHvxmKQWlj0
2ZihWaZWMOsX0yqqF1ILXWrj5xqqXeW1kIphs8HhiOiSNEbUyQHwG9qb4dWbspIhHdLAWjl7MYX1
jxcF8IIZEiABbPpm8VwAuhRyjI5BCCFExB2AEqMvQCZwQCqkQ5I+19QatnxCUB+5rT236EsC+PVs
B/2teGMGFXYbHYaIKhlUVmHp0qUFBQXhr9hstmHDhhkVz3EsWLDAZDKZzf9LQRVFMSrU0lpNoYvg
+S4RfJDzO8F8bD7UILamNG1TUQA6qDcyqWCCRDiynUApPh+qBZuLrLaYLBWdOXF4FgZUKDjxVkII
IeqZg3rXR+jqU+V6GoxOhhPqadEujjmHce/DASYTig1nM5xpeCt3yHggFfx57N8FoJqxOklticUJ
epBx1S4eSg6LIrG3RYsWgtlkMgHBYMV/lYsvvjgS+46wjz76yGq1+nw+QFGUhISEVq1a9erVy+i4
osUU+v/RyLVt2zYnJ0dRFEBVVcBut7///vvDhw8P3+yNN95YtGhRXl6e0+lMTU0966yz7r777mjE
s2/fvo8++shsNo8dOzY1NTX0+vbt27t37261WgGPx+P1el0uV2Ji4sGDB4E9e/Zcd911oY19Pp+q
qjabTftEgWCw31lnTZ06NYJxLoCDep9GeJ9ypL5VgRNt4IBkmDpt7bIPnkkpLThCO3tW+2dn/yWx
FR59qqJTrwedt523JjwR2P9TiafMY0/oce64sfeOVFpSrq8+FocC0BRGGR2GEEKIyPoP7OZ/YwT8
MemWCQ26Buzghs8/5N8vP5ilHiwvL1MTM8c8PPWci20efeaMNl3HDelFfPz4e9u+fB0sZWZnZuez
7pzxkL0phRCo2bIHJ7ygR3xkhwpWuKVuOykvL8/MzES/OfR6vdqTKvfPK1eufO+99zZs2GC321NT
U0899dRRo0b16NGjbgdn06ZN3333nXZ0v//o9SBUVe3ateuwYcOmT5/+8MMPu91uLULA4XCcfvrp
y5YtAxYvXvzY448HAwGT2WxRFEVRTCaTxWLRPiPw97///ZxzzqljwDEmvTGVqKoa+n8fCATCvzFL
liy57rrrDhw4ACQnJwcCAbfbvXjx4qlTp44dO/aJJ56IbCRjx45dsmQJoCjKnXfeGXo9EAjY7fai
oiJVVR0Oh8PhSEhISEhI0N7Ny8tbuXKl9tzr9WqhFhX9rzHCrm8ZKeVGd2KUeHHZ6Ny897JDxUU5
W1D2lRfmrP46d8h1TVQT5aDqZ1grFO7h1x+Xk7MBSsjomPandoEMCoI44mdUrxBCiMahyOhZJWUQ
hC7t+XfOf3PyNrvs5pKDxbu+P//8oVf6bRCWViXBoT18tmiesnOnioKrScaZzZzpuGuQnBhF0fOi
Otb7DAaDZrM5EKj4oA6Ho/o2f/7zn3/44Ye9e/dqbceKonz00UfPP/98KIWotemvvPL+e+8FAgFV
VbX7uuqcTmevXr2GDRvm9XpLSkq0F202W6gVW/PLL7/8uHJleHu9oig2mw1wu93A6tWrJY2p97Ky
su65557U1NSkpKSRI0dqLy5btuxPf/qTqqrJyckejycQCFitVrPZXFhYmJeXN3369OTk5Pvuuy9S
McybN2/Lli2KooR//zTZ2dnTp09XFKW0tHTGjBm//vprYWFh6N3evXuvX79ee/7ll1/+9a9/NZlM
I0eOfOihh5o2bVpaWtq+fftIBakxvC8v0UYQTu1obpPW4kjOT0VqTvCQee3XCy6++mafFUAFLyhB
Mkz8sOhzjhxMoDDZ6s1Nyso+o6fHTjnY4ml6YhVxe4UQQghRF24Vs1KRKsTsVB8+nEx73qkj7Ttl
71r2U5LP71Z823/8TC27EhsBfZyYGRTYtu4AJfkWfCoKSWldB1xoseAGNc6WZwmNhQvFXwDN67BD
LYFRVdVqtXbu3Ln6GJxbbrll7ty5VqvVZrMlJCQ4nc5gMFhSUuJ2u5cvXz5w4MDvv/++1kcPBgKl
paXV7wbDlZaWasnJfffdl5WVFQgE9uzZM336dK3l3a//7n333Re6rX3kkUe++uqrQCAwduzY2267
zWazlZaWdu3atdZxGkXSmKqaNGly//33V3nxoYceOnz4MOBwOEaNGnXDDTcAc+bMmTt3rt1uLyws
nD59+ujRo1u2bFn3AF6bNesvDz+cl5fncrm8Xq/21Qw3ZswY7cmqVau2bdvmdrvLyspC73bu3Dn0
ZNq0abm5ucOGDevXr1/dAzsWE/jAYlBKUw4maNqKTp17/fzrF5ZAgc/r2rdp9cHtNyd3xgwB8EKS
ifJC9qz7Fnc+eIt8nNLrD006UQYR7p+KNHMc5IpCCCEiz4yiX8WCsc1ktMuKNo8/OZVzB1+084t3
A9ayLKftyO5fd/+XFn0qtlS0Cf0F/PbjEvIPJqi+ciUho2W7zv1beMCjTz2Nw7vJUOtkUSTSGLfb
bbVaR44cGboH08yePfvf//63qqput3vMmDEPPPBAr169Vq1a9de//vXbb7/1er3Lli277fbbX50x
o3ZHb9WqVei+TuPz+bSZBcXFxfn5+SUlJYqiaK8Qdov4j3/8Q+s4soTdRnbs2FF7MmLEiLlz56qq
euutt3bq1Kl2scWDuJqRFRe0KVzhFi1atHz5cofDkZycPOZPf3rvvfeGDBkyZMiQ119//dFHHz10
6BBQXFz89N/+Vpfjzpw588knn+zWrdutt9xSVlbmcrm0EW7HmbzkdDq1L27o61uF1g1aWhqtBaDc
cTAzvhwKQGlCt1S5uTcAACAASURBVLPPO1zq93lJSDYV7f1t6wZfQlhBFResW8rOzWvSXUEVyjNa
9L9krKs5FkiL70zGCuVQYnQYQgghIstkqriABsP6RmKgaiEBhX7DBznbdTzo8wdttkMFRZ9+ut4U
tpa0CUoPsm39CrzFEDA5bO07dW15GkVh3TXx7EDdfl1VVa1B2efzZTZpUuXdqVOn5uTkqKo6YMCA
9957T5tM37dv33//+99nn3221+u12WxffP55rY8+ceLEXyr77bfftCf33HMPkJGR0aJFi5deeqnK
L7p1Ho+n+m5DY+S0YgD1V5x/9+LCm2++qSiKoigZGRmz33gj/K37778/Ozvb6/UWFRXt3rWr1of4
5ptv7rjjjkmTJm3atMlms3m9Xm0EpDbCsta71b6m1ROzSPkdDP/6W/Qb/RZ9etCqdyIpVp8v6Dm8
b+2nKXo5SK33/MelXwWLcoNmny0jtWnb0045o7kDMiCg94lXecQJbX3PvUaHIYQQIuJM+uUmlg2C
ocJi2uMIpLTjtF6DIdlT6rGqZXvWLvKXUqp3ECXCzvV7Dv6+DsrLMQddzdp0628yEQCr3g5oAZu+
gmcwbJ2D2KvyX9IUoWqfycnJiqL4K9/0BwKBbdu2ae3Ot99+e5VfGTlypKIodrv9qIlE3b311luA
zWabNWtWlR4b9NkvwFFvI0PVbrUp/vWXpDEntmfPHkVRvF7v0KFDq7/bsnlzwGaz1eVrqk3ZDy+C
4fV6tRlXxxFKpo9DVdXw0syRVVh5eqIhOUAiOMEPTU+lZZfzLaoDNUBJ/n/XLNLe1S4Sh3LY+vN3
eIuP+ErKSGjf67zktijghHLwhWUygDeelpHRRhrkGR2GEEKICNLuGEJX8VDd/xhcRkOLV2oz4K2Q
4OCUbufZUzICqt+ilu5dtSB3Z0VBswTwwsbl88jbDuXlWJu27tZv6OlAgj5tJiRUbM2i799Y2mi9
iFQi1XotqsxR+f333wcMGDB48OALL7ywymAzoLy8XFEUn88XXm82Uh6bOHHbtm2qqt5yyy1Dhgyp
vsGxxuk0MPU7CYuNdu3aaZnA+eefX/3dXXqSk5ycXOtDjBw5cs2aNdoYMEVRZr722vvvvVeTwWBa
T8uxemzMZnNdOnNqwqQ35xA21pawk1cM8gEFfGBP4pLR17+27B3UAvD9vuPXX5bR4RxKIBHWb2DP
5tVmxR0AHE3a9L2CZArAR0W15SBkgA3cUAyJemNS6Bpj1Ng5U+UlxoQQQjQA/rB0grBLTDDml1Gt
seyMoect+bi1e+dBPwQPHvzxP6tGdOvrhQQ4+Ds/rV5uUv1BwOHMPmNA514cBouet6hQXDk18lVe
dsYo2hU8eg32bdq0+frrr4/17oIFC7R7sG7dukX2uJs3b54+Y4bb7e7fv3/EK+XWL5LGnNjHH398
rLc++uijvXv3Op1Om83Wv3//uhzljDPOCD2/87bb5n78sdZFw7G7XILBoDZz5vglLKKnqPKURO0U
Fgolls0wWlXlLv2aJbY6teT3n8x4jpTuW7Hgy9MGXIgJJ2xd86OpaK+dgjK3iimtdZcuCWa06YkJ
oT+DQ1is2GykOCkDh54gKYaejsNTRCGEEA1D9RVAAnqjVQwuo2rYYDYfeCCtC8079f5t33qbVfEn
K+u/+de1d/QtsWGBzT+UHdm1DZsNm5Ok5l36X+jTl2RJAKd29XdTkkvQjjMDkwKGrsZm0rPBUMmy
mCkrK/v99z0rV66aPHny3r173W53586d33333cge5YYbb9RWbL/11lsju+d6R9KY2tu+fftNN93k
dDp9Pl9WVtaDDz4YqT3b7PbQSLC6//nVZOxZ7YQvPFz9PjvGRRjNkNmGDr0GbjywKUkJePyH1nz+
yaV3XJjYjiN5/Lrq66C1xIoXNalj17PadcQOFnBBADauY+m/Fu5c9TO+YEbLrL6XDDnzio5eC36w
6MvOBKKwMlcNP5dJemOEEKLBCa99fFTRvoyGBoD5gTROPWPg+i/fQTGbSgv2bFm9ZzXN+2JS2brm
c8pLTWrApNL8lO5te6WX6GlMOhTvY+7ba7b+sNyTl4eVtr36XHTtiKa9sNgoj2bwx2c6xvNo69Wr
V1lZmTbpHzirX78fli+P7Nj+2bNn/3fbNpfL5XK5xo4dG8E910cyN6aWlixZcvHFF5eUlGj1vB99
9FGjIzoKk8kUvYkxIeZqo2NjTLsS5IOSRNf+F1gTEq3BMpu5fO+mlbt+QYX929i9ZX2q2W23mUh1
9e8/qHUWNjeOIL4gKz4vf3fCzStfGp+77p+5G9/d9MGTix4Y8Y8nN5R6jR/Xq5GuGCGEENETBI9C
5z4DUayYrcGSAt+h3dtW/tdp5dAedmz4noAP/EHF3LZTj8wOFEE5WKBwO6/fPf3zR+7M/fr13NVv
u79/e8MrE6dfN2bPvwq0uamNbemzFi1a5OTkOBwOl8tls9kK8vMnTZoU2UNMmTJFWw3z+uuvj+ye
6yNJY2rj0ksvvfbaa7ds2aJVEnv++eevvfZao4M6Cm2VzOhlMuHVlk36P81hs2XqPtbNfKKHSZ8T
qc1pafKHbl6rze9TvYX2VHPZ1tU/pKrsWfpr8PA2RS08WBKkSedTLxherOC14zJRtoUFT9+6b+uP
ZLQ4787Z101e3OGi63P3//bLCzf99uq2VvrKmAbmadq8T5dBRxdCCBEN4RNglLAXq1w3YzNkXEs2
2p3taNLzQgp+hyzLEdcv33ziy2ffBtybt5lUFQiWeHsMuMDlxA4Z4ICFc5ctXfqmvZnScuAfRz/z
/injHi1smp6z99cPnpkQcFdcv0K3ICe8oEepzS6W445eeeWVd9555+67705LS0tOTt61a9dzzz3X
u3fvSO1/zpw5ubm5BQUFrVu3/vvf/36cLbXZ0YqimI52E2gymbR3tbVl6i9JY07O9u3be/bsuWjR
ovz8fJfLlZ2dvXLlyvHjxxsd19G53e6oTpsJDT8NVY43V05jokrR++JVvdaKH9I60rxb30JvskKy
WSnbvXm5/yArvnzfXFaQV+jDkdb97GGZnSkDs4lgkM0rD+/ftg6f/7LbH7/lifMuu7n9XZMmdex1
TqJS9N/v55fkYg37LIb0imgTKGX0pxBCNCRK3NyBqfr8Umca2f0udLlapNqSwLN90zf+IvZuWubP
3wt+k4rjlOzsM9qZTKhQCjk5fL3kM/C36Xb2/3t58g0TBtz/8s3Dr7wq4D6Qn7MpZz/WypVyGoOu
XbuOGTPm6aef3rNnz9ChQ7V1Y3799ddI9clMnTpVa5uuY9mAYDCo3R/KujGNyKRJk84777wdO3YA
qqrecMMNv/zySwST7Ij7v//7vzvvvPPcc8+N0v7NYQmMgULLDJsg1UWP/sNJa55ot1ps5l1b1n7+
4U8b13+jql4AR8uufS9omlYRcGkJa9YuD7qLE7Oyz7roIhUKoHkXWve4pMRt3bpla3HJ/1IXoz5j
QP+AQgghRN1p17UqbZzaEmpX3zyi1JnpcwZNjqJ9h39d9ul3G1e+V+reBd4AlrandM/sUDFUzAP7
D1B0IBdvcuszr7B3YC/4E2h9SmcIHKb0YEHFwm6N1qxZs7KyskpLS71e7/z58yOyzy1btuTl5SUn
J19xxRV12c+IESNuvPHGG2+8MTs7OyKBGUUaeWvqpptuevfdd71eb0ZGRlJS0uuvvz5s2DCjgzqB
CFYdOKpmsDesj8KoFhetF96kzzjsde5FSz943b99ucXBvp3r33j2dgIHApRja01Kpy79enn08W9e
lfyigya1OC2jRUISbnCDYiG9WU9IzD1ScOgQbdob9Kl02vKXthNvKIQQot5wgaVyxTCjhBrp/JDZ
hnad++/89Qu7LaB6i995/i5TWUGSvfSwN5GUJiP+fJvNgbakXWEQh5M/jp2Ar7zPoJ5BK/4gjnK2
/LwaCsgckJhaMXAuNP7cqNbAzKjteciQIUuWLAEuu+yyf/3rX1XeTUxMTEpKysnJadq06fbt2+t+
uHHjxgGqqqanp1955ZV12ZXL5XrttdfqHpLhJI2pkTvuuGP+/Pnp6em5ubmdOnX64YcfjI4oLjgB
Ksp5xQPtRHnG+TTvcHrBthWAk+LSkoKgkguJqGmn9x7SpC0lYb9iphzKTCqKHxt4QQmg+O0oVkAJ
VIzWNfACE4SEaJ6FhRBCGEXrEgnojXHK0VaSiRkPZCXRecAFO3/+xmxJtLhLC3ZtaAoJqU5sCc06
dj6tb7cyvcky0UTbTnS9v7MXLGAKklbOx9NXrPvuU2wdep51Vcs2+MFfeZnsWApNy7FH7RDJyckO
h8Pr9W7durX6uzk5OSUlJUBhYWGHDh3qfrhvvvlGW2m9adOmjWR1yxOKk/vPuLZy5cr58+ebzeac
nJyOHTvWJId56aWXiouLgfT09Ntvvz36MRojtCytIcvWaAetPtoqqHD20NH/Wf2Bz5tvo8xsLilW
A0G8KLbm3c+2JOHWT2pmUFADqmoJYA1iARskKJjNmJVA0FSxTaTKFdRaItR+aVUhhBDxSgVf3MyB
LIPD0K7vMN6aVlbqsdt8Tb02Fe/ewlJSMrr0Oq9VR7ZRMVTMScUQhnLIhJK9vPSXV9Z+9jZ4WnU/
56b7xyTaqLKGdzCG+Uz4ujHRq5Fz/vnnL1261O1279279+mnn544cWL4u1OnTi0uLlYURVXV8Kks
Tz/9tN1u93g8LVu2vOGGG2p4rNzc3NzcXG09zcxMadusYPhfTT3w1FNPFRYWAjabDbjjjjsSExNN
JlMwGNRq3gEulysxMfEvf/mL9ivPPPNMXl4e0KFDh4adxtih3Li2FvQeeSWswIsKp57VYYGzlVKS
G1QCxbiCagHYaJrecWjPMuV/JSDNZkoKy1GbWK02rV1DqxmQY8kPuMoSEklQKv5CDJyhmBrNliQh
hBBGKfPit1EK5aAYd0OmXeC0y2gJZPaHpBYO3y6P233IZsObDinOYMs23Ucf1IcnKGAHN/jB6ufb
xe4PX3gsd8UPpKT0GX7HdfeOdXXADSqYIaA/YtN9UP1upGXUjjV+/Pjp06f7fD6PxzNlypS8vLxn
n33WarWWlJTcddddH3/8sdvtttlsmZmZoYXUV65cOWXKFJ/PV1JS0rdv35qnMdu3b9e6YtLS0ur7
hJYIkjTmBLZv375hwwbtuaqqubm5s2fPttvtWm0Hr9erpTGZmZnhaUzj4YRivdkjTqqRFEOHHrTu
evr2/RudQVxqaTl4nY7uZ/XL7lzRtaKdr802WnXqvnNlk/y8gsN7adYeFUqC5Oz6Ba/PmpSZnFZR
v8VAFpnfL4QQDVGGjQMAmLRxWYaudGzW5+o0bcrg8y5a8tpSqwsfJdjcqKb01ud169PSrcepjRZT
wFrIF2+smz/1HoqOcEqPa+/927kj2iRkUKwvVKClPQFDx2anRHPnr7/++jXXXFNYWGg2m2fPnv3u
u+9arVaPx1NQUADYbLZmzZq9/fbboe0Vi0UbD6YoykkNDPvhhx+8Xq+iKB6P57SuXSP+QeopqVR2
AitWrMjNzdWeO53OQCBgt9sVRXE6nYmJiU6nU1tI9fDhw9W/jlrvTe2EFnsJBo/5tx8IGJ84OMFm
9AR0tfKIryCUJ9Cyx9nlriYlpjSvmu4lDUeznmf+wems1HVjTqBtpx6mxGZ5BXkl2/e4giRDIIfS
HetwBy2pra0pFSdfA0eUWSWNEUKIhigVEsAENkPP82rYkDYtmek78GJXs3aKq6nJ5jDZzCaHuUnX
PklZ+MGmN+1lQgYsefu3+VMfTCrbmd236wvv//OGG9qkppIGGWCBQNiobEPuV2Jwj3vOOef84x//
SE2tGGVfUFDg8Xg8Ho/D4bDZbB07dnz++efDC8ZaFAXQ1mw5qQOtWbPG5XIlJyerqtr51FNPuH08
3CLGgPTGnEC3bt3uu+8+RVHMZrP2nQg9CWe32y2W//3HvOWWWzZv3rxkyZLwF2suOzv7oYceOnDg
AHDRRRcda7OorglTQ0mQcIwzVIz/gEKT+YJgsdH5omHbd+5MKi9QVI8n0abYU/oMuShgqqjLXLGl
ie5nt0pp17No1TefPHtX0D85vWXmovdfLls7j5bnjrj2dlsGHgjoTUqGCEJTgw4thBAiepLBAgkA
WAztr0AfAKY9ss+ydx16k1fd6VEPmUxBf9DZZ+SfU9tQGtau54Mjm1nz6dvs35qZGezgcm385LOf
5rVUbJmq3em1B7sMTmvdCQV8ei9TjPuaTOCEsjrv54RriA8bNmz37t2zZ8/+4YcfDh48WFJSkpyc
3L59+/POO696WeTevXvfdttta9euXbRo0UllGqNHjz711FOBYDBYw4U0VFU92WSp3pE05gR69erV
q1evk/2tJ554AsjMzExISKjdce++++4TbmO1Wq1Wq8PhsNsNmz2RAgp4jTr80ZihEPqcT59Bt1t8
BAK4TTgsBBWOgD0sJ/FDq05cfcs9b+T8vnHLxo0P/p/J5g/6i6wJLU898/yLRqX7lP9VWTGqrz8I
9XuJXSGEEEfTGjaBVZ/kGTB6DLMmCI5s7nhnNOANYgazCScU6asaAAo4YNXaIzs2rgPLrlLrzmUr
TT/8HAyY8Stgx5I0MvGjjp1s5fpuY3YNNetj2PxQGNH5/cdvO7755ptvvvnmmuzniSeemDNnzrff
ftu+/Uks6TBy5MiRI0fWfHvpjWmkSktLT7xRDTz33HN5eXmjR4+OyN6OateuXSUlJaH5OYbI0Puj
je8Y0nnBB4cgxYTXBnodmFBbV2jlLwuoNoaObm5TZi16Z8bu/bv8nuJ2rbPOPufsC/5vpN2JBzB6
vHIAmhl3dCGEEFGSpjcCKvrNt7FM+jCwYj0TCJpQtWXWwFd57FnAh5uEpr3Ox9+93B+wEbQTANVs
VoKmRJMlyZFqKwWvETM8tS4grY21ed33piihu6z9+/fXeX8Ab7/9tt1uv/zyyyOyt+r27Nnj8Xi0
yIMNOp8xHWfqRaPStm3bnJwcwGaztWvXzulyAf949dVadMUADz/88Pvvvx8MBn/99VeXK5K1/jwe
z5AhQ/yqCuzZvfvAgQPa5BxtBJohPvJRbD1KsTK/EcEoNUiolLAnoed7DpKXS4KZrAyyMilX8Ojl
GivKmkUl3hNLgasNOrQQQoiomgNHwKbPmw+/34yTm7NQFVAtHq3vyAZJPopLKCjEpF9Hg0EIEAwA
mCyQSYITa9iHik2boDZZtxgsYIeRdV54zePxNGnSRHuelJTUpm1bwGwy1XoJwT/96U8LFy4cNmzY
vHnz6hZaVXPnzn3+hRdUv9/tdu/fv9/n83m93jPOOKMBr3YovTFVmc3m7du3u91ul8ul1VmuhQ8/
/LB9+/bffvttREMD2Lp16/Lly202m9frdblc2qSdiB/lpCRYKIuDZiRNTTqFQqsmawX7tUQloxkt
m1XUYCkKu34EwpKZGM+Q0VKypBgeUQghRCy1oKIwcfzPYAj1F2nTY3xWXGkkpuHTNwiE/bSAD8r1
ZT01sRzXoN0M2CO0eLR2oxUIBPLy8kKtxoFAoBY3YFu3blm8ePEVV1zxz3/+MxKhVZKbm/vjypXa
c4fD0eAnxiBpTEifPn206nhaCbzi4mKn09myZS2rjf/2229RWmC1XZs2w4YN83q9gM/nM5lM2gyZ
aByrhlJM5IUVXK4Xxe9C1wwtWq3L26qvQWbSM5bw3Cw2OUwov1LABmXQJPoHFUIIYYjOsAu8cZzG
BMOGOZj1cRYBKurfhOcwJv0JcXAnYNGn+Ned3W7v16+fNq4stFogNZj6f1SdOnXOz8+PRFxH0bFj
xwEDBjgcjtDChlVW3mx4ZFCZqKvDsEjFp1Tc5Yd6jQ0ZVFZzStjP0CuhLCL0CGUysTkphwIAEqAM
bpLlL4UQouGaA/ngiLNZplWEN+qFrobByhscNYEJhr0Sm8uoVr26BJKgA5wTk4MKo0hvjKirTEhU
KAnLW4KGzomvIe1qocUcijZ0UjZV/qchisEpOYwQQjRoTSA3jhOYKkLFcpTKF/pQnhNa7FJ7xZCb
Ae0/phVqM7lZ1CuG9/uJhiADzEbf9NeOSZ/6Eqwcf1BPxsyVKwHETBCCXtrG/LhCCCFiqdWxl1+L
H4r+COodLFWumOG0O0uTcXU+tdVvXBGttizik6QxIgI6hdWLDB8dW1+YjvHQBGM+sx8wQ4pC1xge
VwghROx11lfADIRlC/E2VSZYrb2vegugNihOW7DF2GZNMyRAF0NjELEhaYyIgNbVViCO/0Fl8el/
FzCVNCUCBe+FEELEObve/x+eFcRnPhP/tDEUdjjF6EhEDEgaIyJDa0xSKw+KFTWhVK4uANggUyHZ
yKCEEELESG+9Qmb9FdSHch3nERtmsEF6rA4njCVpjIiMbnqhEvlKnZTqPe+havcDYx2LEEIIA5wK
SfooBu2nqi8AUF/EOFcJqdJnpT1xq3SOeSTCEHLPKSKjE1hAW7/GXK2GiaghbWyxGVKkjKAQQjQa
zcEcNr2kPpbMMUqVXiwzmFTaGRKKiDlJY0TEXAr+sBxGvls1oY2+U/T1xXygQAL0MzguIYQQsfMH
cIAPgpU7YeKzQ0brLDrqI8YBa0tFm/XcTwELtLTFNghhHGnwFRGTCc2hRG9MErWgdcqny1+mEEI0
Ms2h0OgY6qmAPrnIBk74g9HxxFJxcfH69esLCwszMjLO7NvXotTrOVYnTW6WRCR1gzV6Y0x8tiHF
OQs4oKPRYQghhIixs2AvuMEHgE0uozUQGoZn0guUNTM6pJh59913Z8yYsWvXrkOHDjkcDrfbbbPZ
unbt+s4773Tp0ljKTcvAHxFJp1JRX8tvcCD1Q2jwQKjxJAEyoL1hEQkhhDCGAzLBq19AzfW8dlnM
aP/FTPqQ7L5GxxMbzz///PXXX//zzz8XFxcDbrdbe33Dhg2DBw9eunSpodHFjikYlOE/IpLWwzoo
BlvlIsKiuiD4wKG3urnBCedJGiOEEI1SIcyHIrCAE3xyAT2RoL7gpgVs0KNxpDE7duw4//zzPR5P
Xl5eWlragAED2rRps3Xr1jVr1pSVlQEdOnT45ZdfjA4zFqQ3RkRYT3DqJcs0SrU5i0JjCqsUqRVF
cEoOI4QQjVUqNDOocnH9pS1ZkwjNG0cOA9x+++179uw5dOhQixYt5s2b98knn7z88stffPHF3Xff
bbVagU2bNt11111GhxkL8pciIu8PYHVXLCOj9cm4oczoqOKQAg5Q9ItWMowyOiQhhBAG+iOkgRnK
pPmvBiyQqD8ayVoxxcXFa9euTUtLAy666KJzzz039Najjz7ar18/t9utKMoXX3xhXIyxI2mMiLxM
6OjApNcrU/RTjKjOBl7wgQnOksV2hBCi0esATkiQuTE1oIIHFMiIg+o4gUDg0ksvvfLKKy+99NLH
Hnus+gYrVqy49NJLr7nmmksuuWTDhg21O8rHH398+PBhj8eTlpY27pZbqrx7zTXXKIqiKMru3buX
L19eu0PUI1KpTERFf/gdvACYwKGXE6lCmpp8UA4JkAqnGB2MEEIIw50NJZAPxdXeqp7YyGU0AHbo
ZnQYgNls3rx58/bt2x0Ox+bNm5966qkqG0ybNm3x4sVer9fpdPbo0aN2R/n0008BrcvlzD59qrzb
r18/VVVVVU1LS1u3bt2AAQNqd5T6QnpjRFTYoQckgAV8+iKPogoflIEfrHCm0cEIIYSIE0Mho/Is
U6Rz5mgUcEKHuKmzPGbMGFVV3W53Xl7eypUrq7y7bNkyu90O1DqHAfbs2aM9SUw8yjCXVq1aZWVl
uVwuj8ezY8eOWh+lvpDeGBEtPaEE9kEhBECpdkZutIIQADN44QikQCdobXRUQggh4kcLOGh0DPHP
Ci3jqR3w8ccfnzJlSlFRUVlZ2QcffNCvX7/QW6vXrHG73QUFBQ6H4/HHH9defPHFF9etW6coSmlp
KRAIBILBoMlUMcDcbP5fZ8Pll19+9dVXA16vNtKFjIyM6gEkJibabDbteagKcwMmaUw9sGjhwhU/
/pibm5uYmHha1663jBtndEQ1NQDecRN0VKQx2t+lNmHGrOc2mgbfLR76pP6wD2uDFGgBZxsTlBBC
iDjVFfZBAPz6CG1VOmT0lS7RRxOlwgVGhnMUgwcPnj9/vqqq333zTfjr8+bOVVUVaNKkyZAhQ7QX
V65cOW/ePO3140tNTdXSmNDGWsdOdQkJCdpmWvHlhk3SmLi2aNGiRx55ZO/evaWlpV6vV1EUVVVf
nDLl8ssv/9vf/mZ0dDUyzMHCIF4TprD56ya9QuIJB5udcANFz4vin6InMCqYwAJJkAbDDY5LCCFE
PBoKn0EumKqtldw4BfXhDBawgAn6nfiXYu2aa65ZunRpcXHx/pycFStW9O/fX3v9008/LS0tdTgc
Wjaiadq0aadOndxud6gLReuBCQSq3v40a1Yxbs5kMimKQli3THVaqtMYVoaUNCZ+zZkz55577jGb
zUVFRaqqOhwO7Su7ZcuWSZMmbdiw4d///rfRMZ5YE+hpYgcUBzGbKtpRtJ+NbcJMqLFFWyLGDk7o
ZGREQggh4toA+ASABKDxXTersEACeMADdjgVsowOqborrrji4YcfzsvLKy4unjZtmpbG7NixY/Pm
zYqipKen33777aGNp02bdrL711aGOQ6PxwOoqup0Ok925/WOTPGPU/n5+ffcc09hYWFubm7r1q3v
ueeeOXPmTJ48WZsW5nK5Vq5cOXv2bKPDrJE+cArYfBW376E+GXPYIl/K0R40uJYn7cNawQGJ0AGy
DY5ICCFE/EqBgZCqtzorYSO0CXulgV0rj0NrAVSgeVx2xWgGDhwIeL3eefPmaa88+OCDWg9Ju3bt
2rRpU5edW61WrTemqKio+rs+n8/tdrvdblVVk5OT63KgekHSmDj13HPPHTp0SKuaN3369BdffHHk
yJH333//MQjHZQAAIABJREFUihUrOnfu7Ha7S0tLp0yZYnSYNXU6tLFVJC1K2E+N6WgPrSs02FDO
zqGuGIvenpQFPY2MSAghRD3QAbqDAjb9lUZ766bdG1ggA4YaHcxxXHfddQ6HQxtE89VXXwFLlixR
FMXr9V566aV13HmnTp20NMbn81V/9+f16wsKCrTnXbp0qeOx4p8MKotT//nPfxwOB9CvX7/hw/83
eyIxMfHJJ5+8/vrr3W73rl27fvrpp969exsX5kk4D+aAH9ATFa1zPDRyU/tnwz47h9YDbQLnnmBb
IYQQAuA02AE7wGV0JDGm6sOwNUHwQDJcbGRQJzZo0KDWrVvn5OQoijJz5syysrKCggKbzZacnHzn
nXeGb5mXl7dp06Yyt9uiVLTZhsqUVZnZcvrpp6enpwOXXHKJVhUgNzd36dKl555b6W5i5YoVDodD
K/rc4BeNAUyNYQJQZP35z3/+7bffHA6HGb7+9tsq75aXlw8dOrTc6zWbTMOGDXv00UdrcYj9+/cP
Hjx49+7dbrd71qxZN998c5UN2rZtu2/fPlVVx48f//LLL9fug8ReEBbCESiCIFj1Cfpm/d1g2Nkq
oCc2puPWMYvbKf4mvYaBX29Fs8IRcEE6jDA6PCGEEPXLMtgOfgiEXRYbdsFP7UNpBbnKwQ5tYWBY
x1Tcevjhh5977jmbzdakSZP09PSNGzcqinLmmWcuW7YsfLMrr7xy6dKlhw4dCk3xV/R8pkr5sptv
vvmVV17RnmdmZgYCgYKCgjFjxrz33nvhmw0aNOi7775LS0tLSUnZuXNntD5e3GjYbd9R0atHjy1b
tqxatWrlqlXz58+v8u7rr7++bt26/fv2bdmyZeTIkbU7xI8//rh//3632+1wOI7a2dKyZUtAUZT/
/Oc/tTuEIUwwEhz6zb1P75wJ/0ttMPMXg3omo4blY1ZIkRxGCCHEyTsHssAOCRDQxzWo+qNB0j5a
OfjAAWYYVB9yGOCpp57S1nXJzc399ddfk5OTvV7vww8/XGWzpKSksrIyl8tlP5FAWK/D4MGDAZvN
tnjx4kWLFoVenzhx4oYNGwAtw4nF5zSa9MactNzc3N69e+fn5wNnn322NuoxZMiQIT/88IOiKAMH
Dvzss89qd4gpU6ZMmDDB4XC0aNHip59+SkpKqrLB8OHDteMmJyfn5ubW7igG+hAO6Atiapm0OWwy
jPZKfe+NAWzghzK92L8T2sCFRkclhBCi/loGmyEITlChYa8M4tXXzHFCU7gA6tGk9SuvvPKLL77Q
xnelJSenpKdX7x7Jy8vbvHmzz+cLdcIcq+Bynz59EhMTtecej6dHjx6///671WpNT08/55xzWrZs
uWvXriVLlpSWlqqq2qNHj9WrV0f588UFmRtz0po0aZKcnOzz+Xw+3/fffx/+ViDg37p1q1b/+49/
/KP24pIlS+6///5AIHDUyVjhWrdu/eWXXwLFxcXaEjFWq7V6DkNYuT2LpV7+HxwNH0GBXm4lELYm
ZgOjfa4ApENqWA4z/qY/q4p1xmuvGxaZEEKIeugcSIQdYIHyhp7GlEIyKNAMLqhvU4OuvfbaJUuW
aEtlWO32Sy65pPo2GRkZ55xzzsnu2W63T5o06YYbbigqKvL5fO+++67NZvN6vRkZGW63u3nz5v/8
5z8j8AHqg4Z33xgLt91226FDh3w+n9fr/fLzz0Ovv/LKq4cPH87MzExOTr7++uu1F3fu3Pnzzz/v
3bs350Ty8vJCu9Ly8tBMryrMZrOiKKHcvT66Ctro61iFqpNpQp0w9Z3WiWSDLGgBl+mvlx458MOq
72bPeatZ27bfVpteJYQQQhzH6dAL1CBF+miE6ncDDaAQswkcAKTXwxwGGH7xxYDT6XS5XB6PZ9y4
cRHc+ahRoxYvXnz++ec7nU5t9FpmZqbJZBoxYsTq1as7d+4cwWPFs3rZlm+48ePHP/3003l5eYqi
zJs//8Jhw7TXP/roI7fbrShK9+7dXa6Kv7iuXbsOHjzY7/efsOekQ4cO2hOPx6Ol78daovVY6U39
MgzmweHKGYtJ/3n8sWTxzKQPdQuAAnZoA+eFbeBMab5648YLR4/76ptll19++fDhwxtPw4kQQoi6
OwUKTITWDakyuLrKwjJxezE16csqVIlQe8UMDmhebyeUWhQlISHhyJEjiqKkpKR07949svvv37//
119/DWzatCknJ6d169adOjW6JbUljamlPn36LF26NDExMbw1fcWKFYCqqmPHjg292L9//yrzZ06o
RYsWgM1mqz44UqMt0Qrk5OScZODx5QpYB9vhCBRCOthBBW34nXYiCx630zB+JsZoZ2GtOpkD/JAO
wVK6OaneKmIm+T8ffvjctGmPP/jgwoULmzdv/o9//KPWNSGEEEI0NmdCNvxLJUGpmAePPpwhfGU2
Nb4zmVDBUu26r8XpgiAkQ+f6vMDaxIkT8/PztfboK664InoH6tq1a9euXaO3/3gmg8pqafLkyS6X
69ChQ/n5+Vu2bAHeeecdp9OZlpbWrFmzW2+9tS47T7DbtdJ7WkHx6hscOXJEe9KxY8e6HCgenA4D
IR3S9JLwoSlE9ahPXNW7X4KQAIAdkn1cdrQcJuSB8eOLiosHDRpUUlJy1VVX3XLLLTEKVwghRP2X
CjcqNAcFEsAP5rDaOcR38c8qo0pCuVYCJEET6Fk/c5grrrhi+PDh/fr3nzZtmqqqiqJkZma+9dZb
RsfVMEkaU0vdu3fPysoCfD7fc889ByxbtkzrPLn44rquy9T3zDObNGni9XqLioo2b95cfYPCwkLt
Sbt27ep4rHiQBZdDB7CABzxx3G50HKEpPYlg8uOCC6wnzsEsirJgwYI5c+ZkZGR8OHdumzZt5s6d
G/VYhRBCNBR/gA5gBgtY9TXZzHr7WtyWYw7q4aGnNNoCa0mQAZdxvEbAeLZixYrPPvts+3//Gxo4
c++99xobUgMmBZdrb8KECa+++qr2fPHixddcc01ubm5KSsqKFSvCs4tdu3Y999xzgWDQcrQZ+eGz
XFJTU5966inteadOnY4cOXLo0KFx48a99tpr4b+ydevWgQMH+nw+VVVnzZp11VVXRfyjGWUjbIay
sAakuD3/VqeCFZyQzP9n787Doqr3P4C/Zw4zDMMOgqiJW5KaGuaWaZbpNbXM0q6pddU0y7Uyw0yz
m12vS5ZrmntqRWKpSe4/FeWK2zU1SCXMFRcEAWUbmJkz8/vjdLgDAiICZ5b36+HxmWYO53zGp2eO
7/l8F7QDat7/GV4Z0P8/B/6TfedOz549N27cWPklEhGRk0oB4uQ5mRYgGzCWNMaskKWKezXSv3jK
cwd3k7/E1AH+QBjQtCoLq2qhoaF5eXkmk0mv13t7ew8ePPjjjz9WuiinxRhTcefPn5cmbImiWK9e
vatXrwqC0Lp162JrT33zzTfvvPNOTk7OPU/YsmXL3377TXrcu3fv3bt3G41GLy+vxMREab9LyUsv
vbR161ZRFBs3bpyUlFSZb8k+/Ae4BuTJexU7Cm9ABzQBHmSA6rYd2z54/4Pr16+HhIR89OGHQ4cN
q7T6iIjI2cUD14EcIFMepO2haIyRlBhmBLkhI2UYM9AG6FiVJVWP27dvp6SkmEymunXr+vn5KV2O
k+OgsoqT5qUIgqDVas+dO2cwGDQaTYcOHYodZjKby5NhABgMhsLH7777rjQ9RqfT2S4YMGzYsC1b
tgDw8vJq06bNg78LO/QU0AzwBNxspsdIn8JqeR2wqlbYiJd+Cr/Q0gJauQDB5hnpO6R+D5ZhADzf
8/mzZ8927949PT198scfP3HX/05ERORq0tLSjh07FhkZOX369Ly8sraKaQn0APSApzxJRi3/U88s
R5fCIdBClc1BLfuEgs1tVBoCJxjRzikyDAA/P78mTZq0aNGCGaYasBvzQObMmTN9+nSLxZKTk+Ph
4WEwGLKyskrcsLICXnvttR07dmRmZmq1Wj8/v8DAQFEUk5KSPDw8ANSvX//o0aOVdS37dBPYD+QB
boBZHuYrschT6lE1Wdx2wK5F/qjVAdK2o1KnSMowbkBN4GEgtFIL2Llr18dTply+fFmtVs+cOXMY
2zJERM7uv8eP74+JSUpKysjIuHXr1qVLl6xWq0qlys3N1ev1t27dKv+Q4yzgv8AVIE9eCqzwXqmW
U0ThoPbKHb9d7OSF20CLNgfo5J0VAoHmlX0DJdfBGPOggoKCbt26JW1GWb9+/T/++KMST967d++t
W7d6eHiIoqjT6QBIkalBgwZRGza0ddJuTDHJwAm542wApI6VIH8Qi3JLulClfBYX7st5N7WcatyB
EOAxIKAyrliiMWPGLFu2TBTF7t27R0VF8asdIiInFhcX17dv39TUVMi7YIviX/c0Ly8vaZC5r6/v
fZ3zAHANyAWsgBGwAjr5S8Aq2my6sBUj7f9mlO/L0nd/0lpqKsATaMcAQw9G+PTTT5WuwbHt27fv
/PnzOp3O3d19ypQpbdu2rcSTDxo0yNfXNykpycvLy83NTa/Xe3l5vfzyy/v27atTu3YlXsie+QJN
gCAgH7AARkCQZwRKPfFiH8GVksvVNmPYCh+YABNgBfyBukA3oLG8x3AVef7558PDw3/77bfjx48v
Xbo0MDDw8ccfr8oLEhGRYkJDQ48fP56cnGw2m0VRlL5o9vDwMJvNWq12/Pjx3bt3v99z1gcaARag
ALDId0/IW7SZ5REHlUhq+0jfNlrl+yYAd0AH+AE1gU5AW+D+AhnRXdiNeVDt27dPTEzMysoKCwur
3FZMMWZRLHGtM1fzK5AMFAAmm4kr1qJTFR/8/+nCdKQDTECB/NGvBeopMX43IiJi4cKFPj4+Tz75
5MqVK4OCgqq9BCIiqnJ5eXmhoaHZ2dnStomFPZnOnTsfOHDgAU9+EbgM3ALyrMhX/XUn1cjjpSuL
FGC08nd/onxL9QaaAuGVei1ycYwxD2TKlCnz5s3T6XQajebll19eunSp0hW5igLgJHAeMADGKpge
o5UXUTHIyyg/AijbBzkYGzvi7bfT0tJ8fHyGDBnyz3/+U9FyiIioStSoUaNYjPHx8Tlw4IC0OGql
SACuA6lANoAquIcK8ogytYgAAfWARoB/ZV+FiDGmInr37n3lyhWDwXDu3DkAWq02MDDw6NGjdevW
Vbo0l2MGzsufxflAgTz2DPJKjpYy9zNWyXHFYjPBRg34AP6AFggCHranFf0iIiK+//771NTULl26
/PTTT/c7SJqIiOxWbGzshAkTTp8+DXnxUq1WazQaR44a9fWSJVVxxetAMpAK5AEFgBmAvJqZNCSh
cII+7rqHqm3+dCs6Mk0LeAP1gQaAZ1XUTQSAMaZiOnXqdOLECQDSlyWiKEZERHz++edK10VIB84D
GUAuIMoLA2jkBcdsN6KRxuxKswylYWlegC/gBQQADZR8E/dw9uzZ4cOHHz58ODQ0dNSoUZMmTVK6
IiIieiB5eXmvvvpqbGxsVlbWCy+80KpVq2XLlqWmpnp5efn7+1+5cqUaargJZAK3gRQ50uTLkcZq
s/eAxM1mRTI3wBeoCfgC3kAQoK+GcomKrvBE5eXv7x8QEKDRaARB8PPzGzRo0Pvvv690UQQAgUDg
XU+mA5CXbLaNMbqqXGSs6jRt2vTQoUPvvPNOVFTURx999NNPP23atCk0lMu9EBE5pLlz586dO/fO
nTtBQUEzZ84cPXo0AIPBsHTpUo1GM3v27OopoyZQs+gzovznLZsNZ6zymmMCoK/idW6IysZuDJGj
On369JgxY+Lj4729vd9+++3JkycrXREREd2H+Pj4Tz/9dPPmzcHBwX379Ss2cqxVq1YNGzYs50Yx
RC6IMYbIsf3rX/9auHBhbm5us2bNvvvuuyZNmihdERER3du4ceM2btx4+/bthg0brly58oknnih2
QExMzMMPP8xpt0SlYYwhcnhnz54dM2bMkSNH9Hr9uHHjuIgZEZE927Nnz4QJExITE+vWrTty5MgP
PvhA6YqIHBJjDJGTmDNnzuzZs2/fvt2xY8cFCxaEh3N1fiIi+5Kdnf3222/v3r0bQOvWraOjo93d
3ZUuishR2c9CskT0QCIiIo4ePdqqVaszZ8489dRTERERSldERET/88knn/j7+2/cuDEgIGDu3Lm7
du1ihiF6EOzGEDmbuXPnLlmy5Pz58507d549e/bd462JiKg6/frrrx9++OHevXsbNGjw8ssvf/nl
l0pXROQMGGOInNCdO3defPHFuLi40NDQLl26rFq1SumKiIhcVJ8+fY4cOXL79u3WrVv/+OOPderU
UboiIifBQWVETsjX1/fAgQNTp05NS0vbsmVL06ZNjx07pnRRRESuJSoq6uGHH46Li/P29p46deqh
Q4eYYYgqEbsxRM4sPT198ODBu3bt8vHxeXXAgGKbEhARUVU4d+7c6NGj4+LiBEHo3bt3ZGSk0hUR
OSHGGCLnN2HChAULFmi1WoPBsGvXru7duytdERGR0/r73/++e/dui8UCYNu2bZ07d1a6IiLnxEFl
RM7vyy+/TE1N7d27d61atf7+978PHz5c6YqIiJzQ/v37W7VqFR0d7e3t/e6772ZnZzPDEFUddmOI
XMjXX3/9ySef3Lp1q0GDBnPmzOnXr5/SFREROQOTyTR06NB9+/alpKS88MILUVFRer1e6aKInBxj
DJHLeemll7Zs2QKgV69e27ZtU7ocIiLHNmvWrJkzZ2ZlZTVp0mTKlCmvv/660hURuQQOKiNyOT//
/HNUVFTTpk1jY2PDwsIiI79TuiIiIocUGxvbqVOnjz76SK/Xjx079uzZs8wwRNWG3Rgi19WjR489
e/Z4eHi0aNHi0KFDSpdDRORIBgwYEBUVBSAsLCwqKio8PFzpiohcC7sxRK5r586dP2/aVL9+/WPH
jj3++OPR0dFKV0RE5ACio6PDwsKio6MbNWo0d+7cP/74gxmGqPqxG0NE+Mc//rFt2za1Wt29e/d1
337rJghKV0REZI8SExOnTp0aFxeXm5vbvXv3H3/8UemKiFwXuzFEhG+//Xb9+vWBgYE///xzo4YN
N2zYoHRFRER2Z/bs2e3bt4+JiQkICFi9ejUzDJGy2I0hov8ZP378xo0br1+/PmjQoOXLl+t0OqUr
IiJS3smTJz/66KNjx47p9fphw4Z99tlnSldERIwxRFTUxo0bp06dmpKSotPpZs2aNXjwYKUrIiJS
0qBBg3744Qd/H5+wZs1+3LChbt26SldERABjDBGVKCIiYs2aNbm5uS+99FJkZKTS5RARKWDdunXz
5s27fPly48aNR44c+cYbbyhdERH9D2MMEZVs9+7dkydP/vXXX2vVqjV16tRRo0YpXRERUTU5f/78
e++9t3v3bh8fn379+i1dulTpioioOMYYIirLsGHD9u/fn5aW1rZt2z179qjVXBeEiJzc7Nmzp0yZ
4uPjU7t27QULFnTt2lXpioioBPwXCRGVZfXq1atXr360efOYmJiwsLDlK1YoXRERUVXZsWNH+/bt
Z8yYERoaOnny5N9//50ZhshusRtDROUyYMCA6Ohog8HQ/oknNkRFhYaGKl0REVFl6tu37/79+3Nz
c1u3br1r1y5vb2+lKyKisrAbQ0Tlsn79+t27d7d/4omjR4489dRTS5YsUboiIqLKsXr16oceemjv
3r01atSYN2/eoUOHmGGI7B+7MUR0f957773vvvtOp9O1bt16yZIlderUUboiIqIKOhwXN/Wf/zx1
6lReXt6QoUO/5hc0RI6D3Rgiuj/z58/funWrj49PdHR0kyZNPv30U6UrIiKqiHfeeefJTp1OnToV
EhKyf/9+Zhgix8JuDBFV0McffxwZGZment6uXbulS5c2atRI6YqIiMpl69at48aNS05O9vf3nzx5
8vjx45WuiIjuG2MMEVVcUlJSt27d8vLyNBpN3759Fy9erHRFRERlSUtL69Onz6lTp4xGY+fOnfft
26d0RURUQRxURkQVFxYWduXKldGjR+t0uiVLlrRp0+bcuXNKF0VEVLIvvvji4YcfTkxMrFev3ooV
K5hhiBwauzFEVAmuX7/++uuvx8bGenp6DnrtNQ4xJyK7EhsbO3ny5BMnThgMhg8++GDOnDlKV0RE
D4oxhogqzZQpU2bMmAGgfv36kZGRHTp0ULoiIiL06dMnOjoaQNOmTdesWdOuXTulKyKiSsBBZURU
af7973+npqZ27949OTm5a9eukyZNUroiInJpK1eubNas2bZt2xo1arRq1aozZ84wwxA5DXZjiKjy
ffHFFxERESEhId7e3gsWLOjZs6fSFRGRa0lISHjjjTf++OMPURT79++/Zs0apSsiokrGbgwRVb4P
PvggKyvrscceu3HjRu/evbmYKRFVp4iIiKeffvrMmTOPNm8eHR3NDEPklBhjiKhKeHt779y5c/Hi
xXXq1JHGdWzdulXpoojIyUVHRzdr1mzdunVeXl4TJ048cvhwt27dlC6KiKoEB5URUZXr0aNHbGys
KIovvvjijz/+qHQ5ROScnnnmmQMHDvj4+LRs2XLHjh1eXl5KV0REVYjdGCKqcjt37py/YEHjxo23
bdvWtGnT3bt3K10RETmV5StWhIWFHThwIDw8fPny5f/5z3+YYYicHrsxRFRNzKL4Yu/eO3bsCA4O
7ty5M9syRPTgDh48OGfOnP3792u12v79+y9evFjpioiomrAbQ0TVxE0Qtm/fvmTJEj8/v9jY2BYt
WnC2DBE9iAkTJrz00kvR0dGtWrXauHEjMwyRS2E3hoiqW0FBweDBg3fu3KnX63v27Ll69WqlKyIi
B7N58+ZJkyYlJSU1btx4zKhR73I5RCLXwxhDRMrYsmXLhAkTzp8/HxgY+OWXXw4ZMkTpiojIAeTl
5T333HMHDx4E0K1bt//7v/9TuiIiUgYHlRGRMvr06fPnn3+OHTvWZDINHTp0xIgRSldERPZuyZIl
zZs3P3z4cIcOHbZv384MQ+TK2I0hIoXt3LVr1MiR2dnZISEh77///rBhw5SuiIjsTmxsrDSKTK/X
9+rVa+nSpUpXREQKY4whIrswYMCAmJiY7OzsHj16bNq0SelyiMiODB069IcffjAajT179pwxY0Z4
eLjSFRGR8jiojIjswvr167///vtGjRrFxcU1a9Zs5cqVSldERMpbu3Zt27Zto6OjGzduvHTp0u3b
tzPDEJGE3Rgisi+vvfZaZGSkIAhdunTZvn27RqNRuiIiUkBCQsLgwYMvXLig0+meeOKJLVu2KF0R
EdkXdmOIyL58//33W7duDQ8Pj4mJadCgwYYNG5SuiIiq27Rp01544YUzZ86EhYWtXbeOGYaI7sZu
DBHZqQkTJsydOzcwMLBjx44rV64MCgpSuiIiqnKbN2+eMWNGfHx8QEDAiBEjPvvsM6UrIiI7xRhD
RPZrx44dkyZNunDhgre396fTpr3FRZmJnFq3bt1OnTplMpnCw8PXrv2mfv2GSldERPaLg8qIyH71
7Nnzt99+e+2114xG4+hRo5577rkrV64oXRQRVb4FCxY0aNDg6NGjwcHBq1atOnDgADMMEZWN3Rgi
cgCnT5/u1atXRkZGTk7Ou+++O3/+fKUrIqLKsX///lGjRl2/ft1kMnG9dSIqP8YYInIY48aN++mn
n1JSUtq0abN9+3bOliFydMOHD9+yZYtGo3nkkUc+//zzdu3aKV0RETkMxhgiciRJSUmvvfZafHx8
zZo1X3rppYULFypdERFVxNdff71o0aLU1FS9Xj9x4sSxY8cqXRERORjGGCJyPO+88460t0xoaOiP
P/5Yv359pSsiovLKz89/6qmnzp8/bzKZGjdufPz4cbWaM3WJ6L7xg4OIHM/ChQsPHDgQGhoaHx/f
tm3bKVOmKF0REZXL9OnTGzRocPz48Xr16m3YsOHEiRPMMERUMezGEJEDmzhx4ty5c/39/R955JGv
vvoqPDxc6YqIqGS7d++OiIi4du3a7du3R48ezRGhRPSA+BUIETmwzz///OjRo61bt05MTHz66acn
TZqkdEVEVIIBAwY899xzqampjz76aFJSEjMMET04dmOIyBksWbLkww8/BPDYY4/Nnj27Y8eOSldE
RADw1VdfLV269PTp06Ghof/6178GDx6sdEVE5CQYY4jISaSlpXXv3v3PP/9Uq9VDhgzh171EykpO
Th4yZEhMTIyPj0/nzp1/+eUXpSsiIqfCQWVE5CSCgoJOnjw5c+bMgICAJUuWPP3008eOHVO6KCIX
NWnSpKeffjouLk4KMMwwRFTp2I0hImeTkZHxj3/8Y/v27TVq1Oj/yiuLv/5a6YqIXMhPP/00f/78
EydOBAQEvPvuuxEREUpXRETOiTGGiJzT3LlzV69enZyc3LRZsy+/+IKzZYiqmslkGjZs2IYNG7Ra
befOnVesWFG7dm2liyIip8UYQ0ROy2KxDBo0aPPmzd7e3oMGDeJsGaKqM3fu3G+++SYpKSk8PHzi
xIn9+vVTuiIicnKMMUTk5ObNmzdr1qzU1NQ6dep8++23Xbp0UboiIqcSHx//2muvXbp0yd/fv1ev
XkuXLlW6IiJyCYwxROT88vPzR4wYsWPHjry8vLfffnvevHlKV+SKjMAd4DaQC1gAK2AGrICbvNqM
HtADHkANQKVwsVReU6ZMWbly5e3btzt06DDniy/atmmjdEVE5CoYY4jIVWzYsGHy5Mk3b96sX7/+
559/3rNnT6UrcmZXgXQgDbgNmAERACACopxhIP9ZjArwBARAA/gCPoAf8HD1FU7l9fXXX0sbwgQH
B0dERIwfP17piojItTDGEJELMZlMr7zyyv79+7Oysvr37x8VFaV0RU7lNGAA7hQgwwCzF8xusADm
oq0V0eZBiS0XNaACrIAI+ABWQAC0gBbQAK2A4Cp/H3QPOTk5r7zyyq5du3x8fHr06LFq1UovL2+l
iyIQCwK7AAAgAElEQVQil8MYQ0QuZ+PGjZ9++umVK1fq1q376SefvNK/v9IVObbLwHngJnAH8ALc
AStgBAqAApvtyQpDi/WuZ4qxyIPNLHKMsQJWQA24A3rAD2jAFo1CIiIioqKibt682axZs+nTpz//
/PNKV0RELooxhohc1LPPPhsTEwOgV69e27ZtU7ocx3MJ+AO4JScW6V4iAAJgAUR56osAABDlB5Ky
t14uPNIMGOWhaNKZPQCNHGwEKwJUaAbUr+y3RiWKjY198803L1y44Ofn16dPn1WrVildERG5NMYY
InJdUVFRU6dOvXz5ckBAwIIFC/qzLVM+ScBp4Bb+GjMmTXexACrADdDKKcUMQG6kFMYYq01KEUo4
NwDoAUEOMPmACbACGkALuMtdGiNgAbSAAIhAc+Dxqn3Trm748OFr16718fF56qmnvvjii8aNGytd
ERG5OsYYInJ1r7766oYNG6QHkZGRanXZrQKXdgq4ABgAI2CWg4oo5xZp3JcUTlSARX7SUnQ2v7rM
hcgEm3hjlVsxhWcWALXc7ZEeSM0fHeANhADtK/kdE+bNmzd//vyUlBQA8+fPHzVqlNIVEREBjDFE
RACWr1jx+ezZycnJRqMxMjJy4MCBSldkd44CF4BsQCXPzhdtXi0MHuqiq5CVcYMpMckUHi/I0UVi
tjk/bDKSuugl3IEagA/w1L3fEN1bcnLy66+/Hhsb6+Xl1bVr159//lnpioiI/ocxhojoL6NGj169
ahWA7t27r169OigoSOmK7EIqcATIAcxArk2EsHXPGFPOfWAqFmNE+QBpTTM9oAVaAY3Kd1Eq0Tvv
vBMZGZmenh4WFrZixYrOnTsrXRERURGMMURE/7N169bRo0fn5eVptdoPPvjg/fffV7oiJeUCMYAI
3AEK5DkqpQ0Jk55Ul9mBKae7R/XdHZxsiTY5RwBUgADogACgORcAuH/ffPPNZ599dunSJa1WO3bs
2C+//FLpioiISsAYQ0RU3LBhw6Kjo00mU8uWLXft2qXX65WuSAFHgStAASDKM2EAWAA14CYfYzuu
zDbG3NfsomIRpYzfLSPMWG1etcr1eAC+VjRQIfx+6nFxgwYN2rt3ryiKbdq0WbFiRd26dZWuiIio
ZJzJSkRU3OrVq/ft2xcWFhYfHx8WFjZv3jylK6pWecB24FcgDzDJ7Rc3QHWv2fnWymjFVIA0lkwH
6OR10qQNNAuAHBFXTdhvQY4ShTmWGTNmNGjQ4IcfftDr9StWrNi5cyczDBHZM3ZjiIhKNWzYsJ9/
/jkrKys8PDwmJsbb2/m3Kj8KnJRXAFPL/RZ10U6I7ULJIqqKUO6Tl7hwsyiPLlPLG86EA2GVWaDz
SEhI6N+/f2JiYnBw8HPPPbdu3TqlKyIiujfGGCKispw8eXLMmDEJCQlBQUFvvvnm5MmTla6oCsUD
8UAGoAH0gAGwFm2/3B0Yig3lqkTljzHlPJsfUAvoVHnndA7jx4/ftGlTXl7ek08++emnn7Zq1Urp
ioiIyoUxhojo3iZNmrR48WJRFLt06bJq1aqQkBClK6p8ccBFK4wqFAAegDuQU0qQsA0zVRdjKp0R
qAn4A92VrsROfPXVV19++eXNmzcFQfjss8/Gjx+vdEVERPeBMYaIqFz+e/z4hxMnxsTE1KhRY+jQ
oXPmzFG6okojAvuAm/JkGGmZL7U8v/9uDhpj8gF3wAeoBTwFaJSuR0EpKSljxozZsWOHwWAYNGjQ
4sWL/fz8lC6KiOj+MMYQEd2HL774YuHChQaDITw8fM6cOeHhDr8IVh6wF8gATIDFZuVilL4IzD3n
xgilbI5ZoqqbXVOMlLikjWWsQDugcXVd2q589NFHkZGRGRkZjzzyyKxZs7p166Z0RUREFcEYQ0R0
fxITEwcOHHjq1CkAkyZNmjlzptIVVZwIbAMygVRAD7jJocIEAHCXD6vcaSoKknKaCJiAGkBXoJ7S
JVWnbdu2ffDBB4mJiYGBgW+88YYzdRSJyAUxxhARVcS//vWvlStXZmdnP/7444sWLWratKnSFVXE
diAZyAcACHLbpHDfFWkTmMJeSoWTTBlrNKPaR6NJl5M2wNEBnVyjJ5Ofnz/+/fcjv//eYrF07dp1
0aJFXEyZiBwdYwwRUQUlJye/8MIL169fNxqNb731lsN9t70HSAPyiuYT0SbGSPHj7hhjvVcyKcau
YowFUMkDzKxAMNAVCKzeGqrZokWL5syZk5yc3KRJkzFjxowdO1bpioiIKgFjDBHRA5kxY8bSpUuT
k5ObNm36/fffO8p6tVuAW1YIqv/tWWkBxKKRptgkeNsYI1HJIafsRo1dxRgpw1jkzTr9gBpA93tN
4HFQycnJI0eO3L59e61atXr27Llq1SqlKyIiqjSMMURED+r69esjRoyIjY319fUdOnTo9OnTla7o
Hr4H7gAi4Cn/890IoOgel7hriv/9xphydmyq/yZU+DbNgA7wsqK2Cs9UexlVbcqUKWvXrr127Vp4
ePhXX33VsWNHpSsiIqpMjDFERJVj1qxZH3/8sSiKLVu2XLx4cadOdrrR4gHgN8AI6AE9oALygTuA
B6AteqTTxxgREAAPwMuKmip0rvZKqsjGjRs/+OCDvLw8tVo9evToqVOnKl0REVHlY4whIqo0V65c
eeedd+Li4tzd3YcMGfLvf/9b6YqKSwDOAllAnrw/jAUwAQZAc1eMKTbOyjkWKwOgkqcAWQE14Abo
AA8gDHhM6doeUFpa2vvvv79hwwZPT8+ePXvOnDkzNDRU6aKIiKoEYwwRUSVbtGjRzJkzU1NTW7Zs
OWPmzB7PPad0RX9JBA4CVsANMJUeS6S7QmmbxjiNwpUMCgnAI3DgnsycOXOWLFly6dKlBg0azJkz
p1+/fkpXRERUhRhjiIgqX0ZGxuuvv75//369Xj9w4MBFixYpXREMwG4gHYA8x93FYwzuWnKtAHAD
XgQcbiniU6dOjR49+tSpU97e3n379ft6yRKlKyIiqnKucJ8iIqpuAQEB27dvX7NmTUhIyFdffdWm
TZs9e/YoW9J+IEdefMxc5ggxtcvcG4rN3rECFuCoMrVU3KjRo19++eWEhIQnn3zy8OHDzDBE5CLY
jSEiqlqdOnU6cuSIKIpPP/30/v37FanhJHAEcJenhRjlaSEluq89YZyJETAB3kB9oLvSxZTHDz/8
MHz4cKPRCOCf//wnp/ITkUtxkW/ciIgUc/DgwaVLlzZp0uTYsWPt27ffuWtX9deQCNwB8uXNH/n1
VYm0gAdgBq6acUPpYsqWkZHRu3fvkSNH6nS6IUOGGAwGZhgicjXsxhARVZMBAwbs2LHDYrE8//zz
69evr7br7gXSgBTAAxAAK2AGLPJjl+29FCPIP2rAE7CK6GevO2JGRERs2LDhypUrzZs3X7Ro0TPP
PKN0RURECmA3hoiomqxfv37RokUBAQF79+596KGH1q5dWw0XNQK3gDzAA4A8JUZV+qb19vpP96pS
mF7+99gKEcgXcFDh0kqwd+9ed3f3hQsXqtXqd999NyEhgRmGiFwWuzFERNVt2LBhGzduzMrKGjhw
YGRkZJVe6yBwCciU94Sx2mxYefeKwygaY5xmo5gyCHc9KFzhwA0YpEBFpfrHP/6xc+dOURQff/zx
devW1a5dW+mKiIiUxG4MEVF1W7169bp167Ra7bZt2/R6/dy5c6voQqeBq0AeYAZU8ie+WO584lKd
GVH+MQNGwAgYgL1KVyWZNm1aQEDAd999p1arp02btmfPHmYYIiJ2Y4iIFDNixIjo6GiTydSxY8df
fvml0s+/BbgOGAC3ooPKJPfsxsA1GjLFCIAFAGAFagCvKFrMpUuXBg0alJSUZLFYBg4cuHjxYkXL
ISKyI+zGEBEpZsWKFVFRUTVr1ty6dWu9evW++uqrSjz5NSDDCgMAm6XJbFOKivP7ZVabH7NNWybb
ihPKVTVu3LjOnTsfPnw4LCzswIEDzDBERLbYjSEiUt7QoUPXrl3r4+MTHh5+4MCBSjnnJiBT/uc4
5I0vbZXYaXG1uTHFFLsjmoCawKvVXsbmzZvfeuutzMzM2rVrv/rqq3PmzKn2EoiI7B27MUREyluz
Zs3hw4cbNmwYFxfXsGHDZcuWPeAJfwMy5PWUNaXMclGV2ZBxwQwDm7+Nwr+ZXOCPaiwgPT29X79+
b731Vn5+/quvvnru3DlmGCKiErEbQ0RkR4YPH75hwwaNRtOoUaMDBw7o9fqKnecX4AagLjoBptjH
vUV+YPu84KoBpkRGwBOoA/SolstNmzZtzZo1ly5datKkyaxZs/r06VMtlyUickjsxhAR2ZFVq1bt
3bu3UaNGx48fDw8PX7ly5d3H7Nmz5+bNm2WcJAPIAVQ2+cR6V4ah8lABBUA68FsVX+j06dPPPvvs
7NmzCwoKxo0bd/bsWWYYIqKysRtDRGSPxo8fHxkZKYri008/vXDhwjp16hS+1KZNm8DAwF27dpX2
u8eBeMBSZnSx2DxmN6Y0RgCAHggW8WKVrT89cuTI3bt3X7x4sWvXrvPmzWvRokVVXYmIyImwG0NE
ZI/mzZv3448/1qpVKyYm5vHHH589e7b0fO/evRMSEnbv3v3NN9+U9rvn5Gn9tvNeLDZbo4hF1+ay
xQxTjAgYgKyqOXlkZGTDhg3XrVsHYP78+Xv27GGGISIqJ3ZjiIjs2vTp01etWpWdnd2mTZvatWt/
8803Wq1Wp9M1atToxIkSVgNOBSIBd3lav5RkCjdCQSnbxVCJRKAAAOADPAE0r7wzp6enjx49et++
fWq1uk+fPsuXL6+8cxMRuQTGGCIie/ff48c/mDAhNjYWgFarFQQBgF6v//DDDyMiIood/B/gTNEV
tyD3WKQ/1UVfojJY5eaVHqgD9Kqk037yySfr16+/cOFCkyZNlixZ0rlz50o6MRGRC2GMISJyANev
X+/cufPt27ezs7OlZ3Q6XWBg4IULF4oduQO4YLNLTGGMKZwqUzjFg22ZexLkjXc8AA9g0AOfMC4u
7r333ouPj69Zs+bgwYOnT59eCVUSEbkkN6ULICKiezt06FBaWprJZJJaMQBMJtOVK1f69u27adOm
wsOuAemAFjDJ+cT2U15d7tCi4spmAACtPI8IgAE4CzR9gLO99tpre/bsSU1N7du376JFi2rXrl05
VRIRuSRO8ScisncWi2XMmDEWi6XY81qtNjY2NiEhofCZM0CuPL/fQdlVd8gN0AIACgAtULzzVW5L
lywJDQ2Niory9PRcs2bNxo0bmWGIiB4QYwwRkb0bM3ZsXl6eWq02Go0ANBqNRvPXqLHs7OyPP/64
8Mgb8iAoAVDLH/FS+rm7FaMqPTNUrBUjAFpAK08pkf6ztGWKBTkkSMcIcj1i0cWglWWW34g0Q6YC
65WdO3euR48e744fb7FYhgwZcuHChSFDhlR+oURErodzY4iI7N3p06f//PPPhISEP//8M/GPPy5f
unTnzh1BEDQaTWZmppeX15fLFr01aCiAtXI3RmsTUYp9yku5oooWVvYBLMAdQAS0gAdgLOVabnJ0
scr1WOQ+kt1+waYDXgL8yn380KFDd+zYkZqa2qRJk7Vr17Zr164KiyMicjGMMUREDqmgoCA+IeH3
hISLFy/eyM1Y8eVXWcBmIAuwKhdjfAEAOUCe3MdwL2VLTYu8lY0AaAB10b6NaJc72OiAzkDDchy5
c9euUSNHXrp0KTQ09I033vj000+rujYiIlfDGENE5CQSgBM23ZjSFKaFqsgJhaElBzDJMcZ2bFsh
izzszSIfqbZ53j5pgUeB9mUek5eXN2TIkJiYGACtWrVaHxkZGBRUPeUREbkUu71ZEBHR/THID9Ry
r+Nupc1UqSz5QLYZ5gJ4mOFlhtqI3LvykkWON7npMGfCLRc+gNr6V/4psMs+TCFjma9GRETUq1dv
27Zt3t7e8+fP/7//+z9mGCKiKsIFl4mInESBPEveAqhK+ZqqqhNCELBtU+aWZe8GiJmCxf22ul7/
BV82fQwmwA1Qy+sXuwPG21g/46eL8bs1BqNV7V2zSdtxMwb71sBt+/6CrbQBDP89fnz0qFEnT54M
Dg4eMGDA6tWrq7UsIiLXwxhDROQkbtlMkZcmxlirff3iAsDXz//26d9zctPdRJNBl3D76CRN8yA3
AQZ5Aowb4A5cPo+z+1bkXvtVpfaw3vSs37Kb3gcF9r1fjQjcKOn5wYMHb9myJT8/v0WLFtHR0XXr
1q3uyoiIXI89f+dFRET3IcdmIJnt/H7bn6qWC7RojUaPttEZNZ5Wo5vlTsLuSHdjkcFsakADXDiZ
mJucgIx0662rCPRo17GrTusAO95kFf1r3PDTtw899FBUVFRAQMCHH3548uRJZhgiourBbgwRkZMo
z4Cxwn+CV1GXRg34+KFeeJcLhw9poAVw4/zv188jsPlf9xsBsABaIP7wXmTnwwvQ1g5s2OLhll72
PCWmkBnIBAKAlGvJI956I/ZwnACPIUOGLF26VK3mN4NERNWHn7lERM4gS54P46boJ7sKEAS06DbA
IrpnGKzmO3cKUv48/2ual/y1mbTC8uUEZJ4/AcHNy6jFHbfmHXq2ePSvrWM08gpmtj/2owaw6Rqe
GfKvJk2a/d+uvVmZ+T/++OPy5cuZYYiIqhk/domInERp0/qruYYCoPbjKq96DaV10axpqb/H7THI
w8UsQC6QeOzUlXPH1aJZbRS0oQ3bdn/ZIK+0dneGsX1TgvyjlFTAZMLFy5f1vv4fTZ1qtVq7du2q
XDlERK6Lg8qIiJyBpegiyyp5/JiqeifNC4AR8KyJFu06H049hTtponjrXPzB3IyBnsGwAgJwIxO/
Htgi3rysFnJE0b1mzYcbNddlAEbAarM+gfRe1PJp776QIoPQRODx+ojbv9Iv3+Sl0yhRAhERAXbw
zR0REVWCC3KMscoBQCXnAVVJP1VEBAyAGeg6cCzg5QPkCqkZl/ZcOJwuAFZAD2gzcCXhBMQ7FoiC
4N2ydR+NP/IAAAKgBdwAN0AE8u5q0WhtIo0iPRmptocAZhgiImUxxhAROQNTKftdVvOCy2a5+dP4
cZVvnYd1noFaAcjNPLD5e9EMDwBW/Hbg+vUbV6TjdXUaNGj7tMkLJsAKaAA9oLMCBpjSoTNCL78F
QV4eQHH2UAMREXFQGRGRkxDt5jNdDai90OyJrglXjrsBZhHnfzt868I7DzWGKh9X4g+psm5bAcBD
W7tJQGNvEwDADNQBLl3EioXb06+eScu4rtIHDHrzw7Y9NH7uf+2bKaUdh1jTjIiIqpSd3PKIiKhy
iGWuqizYHFYVVPLkFsEbj3boeXzbMo3R10c056ecu3G2oFFj95w7SD7+H4vWCK0v7lhqte7aoA1y
ABVQCzj3OxZ9OPlS/CFYcnQWMT8td82JnWdfHjJwwoig+jCpIAJmwKzcRP8qHZJHRETlx0FlRERO
otgel4r8a7twwFUuEN6pXs2QhgL0OqNJk5F+cv8vbsDvJ3EtKR6GXNy5o6rZqGXnF1UCtIAPYEjB
z3Mir+/ZIHj5d3tlwsCxy5o98zpyrUcjl2z/6TfBAosRasAkr9qsCHX1LplARESlYYwhInIGWvsY
bSXKNeQB/g1Ro/lTJq3eqobKmJWWeOz2nzhzeCcyUgAj/GvXCGnySGsfyInr4ln8Z906o9H4ysB3
pi4Y2H9Su+92fvLwI48iK+PiiYOm2/DR/m9RaaHoT7VhK4aIyE4wxhAROYOH5PFOVuXaBbbtID8g
R4uQbkPzBV2uXsxChvHSb2d27Eja962PeAMQ1W6+jTq+EFQbBUA+kAlk6VC3zwst/vF+8791yQYy
BVwVkOelgZhf22LyMkMENIAboLlr4bVqSzJugL66rkVERGXg3BgiIidhD90YablnFeAOGIH2nfwO
hTTMunhTEIy30lO//WZxviEDENSCt8W7RutOfwv0wDVAAAqAJh0w49uxBhX8vJCSDuMl7NuxKfXC
SfgGNuv1d1VNpAFegJui38AJbMgQEdkHxhgiImfgpuiMEVtSxjAAGqBGMBo27ZCQek6dbzIbsm8k
/64qyNfDZDF6+dVuXL9FcC7gBeQDRkAFaLxhAG4Dy5bcOLFqvFt+ktZNfPLlfs2frHMH0NjBe7QC
7ooWQEREEg4qIyJyBv6AxWZqilJU8mJl0gYywT54rF0PizbAZFBZDDmqglSVmFaAApWHf8PHngp+
GPki1PKekpCn73sAderUqtu0hUWjz8vLP/XriQN7Lmbmwg1Q2cEbDFC0ACIikjDGEBE5CY2dLaIl
rVrWvVer4MC6QJ4b8gGDVbAAbtbgkEfaPiXqoBGgBjSAB+CeD7dseJuhA155HRHfTBn15baARq2y
riZtXDjLeA7+gKD0fcsNCFS0ACIikjDGEBE5CS9AAKzKDbtS2dxUrIAKMAJuD6FuyycgBJpxxyoA
gDc0HgG1H2nVqAAwAwYgH4AV+3/Cimlx3359qcAMtRbeIej+d9/W7Z/x0qp1+TevXcQNIyyARdG0
plHu0kREZIsxhojISQQAavnHljQlvdoWJrZNMiKgCUTIo50gasx+Gnh4wjPEQ1cr7NEn6zSGEcgH
sqUYA/z56/Z9aybt+nbqhfNwA7yBWipoAEGF3PzbWm94a5WPMbxrEhHZCX4gExE5Cb08ycROFm+R
JurkAI91aFPnbyNrNhkS+MiggMav6Fu8/ET3NzUeUMsDz1SASoWWTz0DAEkHN/17hiEJqvPYsjzj
7P5Yc0FecN1mAQ8pP/NHbTd/t0REpLJa7WooNRERVVAC8Ku8lpdJTgiSu0eaVXUeEORLCIAGyE6F
vwdEwCpClQ0hCKIO2fJiAALgDqhvY9lHPxz7YRa0Pt6Bdf0Ez6z0lDspyeog3cApa58c/ojVCzrA
CFgBbdHLVc9i0xqgEfBM1V+IiIjuiTGGiMhJ5ACbACugBvKVjjG2ly5cXsxshZsKWjm9iDbHSE0k
MRU7Zu36ectaZN/CnRwE+4c2atJ36LuPvRRq8YMR0AEGwCzFnqJXqZ4Y8wTwaNVfiIiI7okxhojI
eXwvzxsx2EGMEeRWjDShxQyogQIgC9Dhr9WTrXLBHoAVMACBZlw4j/wM5KfBMwT6xvD0h15eTloP
5MoxBnfNw6lqGqA3EFz1FyIionviKF8iIudhFWES/tp60h6INn8WJg29fO+R1lUrnB4jAjqgwA11
H/lrFWaTPITMKh9vu/1lNUzuFOT6cwAtoGOGISKyG4wxRETOo4GAP4F8QC0vUKYg2wwDm+5QsWkt
avlVab1mETACKiAPACDIb0RV9PhqXqBGKq9O9V6UiIjKwJXKiIicR0vAzXE+2W3HNJdnSJhKjhPV
tjFO4SoFOkAPNKqu6xIR0T05ys2OiIjuzRuoIbdi1DY/Sm2IaavEPW3uppIHm0GeP6M4AdADfkBD
pSshIqJCHFRGRORUvPnJXtkEIEBeVICIiOwEuzFERE6lsbwmmMo+WhlVoZpX2LQABUCD6r0oERGV
jd/ZERE5lSCgNpAlL2cMwFLt/+6vGEvRRaIBmIv+Z+HQuGrb9AaACcgHAoAm1XhRIiK6J3ZjiIic
TS2b6SXORJR/qpM0n0d77wOJiKhaMcYQETmbMHlyvNTN4Af9g3AD9FxqmYjI/vDuRkTkbIIAT8AN
MFv/GlrGz/qKUQE6wA9oqXQlRERUDG9tREROqDlgBrTOOse/KhX7O7MCjyhTCBERlYUxhojICT1s
s0awPWwa41ikJKMG9EAg0EzhcoiIqASMMUREzukJwODsKy9XKSvgCdRVugwiIioRYwwRkXOqCXgD
cJwFlxWntrkpCoAKcAdaKFkRERGVijGGiMhpdQS0gKXaFyl2dBZ5nTcuUEZEZLcYY4iInFYo4AdY
5JWXbfdd4YSZu0ltq8KJMW5AuMIVERFRqRhjiIicWRNALTdkrPKPhEmmGDVgBSzy44cVLoeIiMrC
GENE5MwaAA0Bj6L70AvMMCWxyssheAEBwLNK10NERGVgjCEicnJdgECb6MJVy0pjkX/UQDeliyEi
orIxxhAROb/WgB7QyElG+pPz/osxAQZACzQG9EoXQ0REZWOMISJyfnWBhwBfwI3DyUqnAbyBGkA7
pSshIqJ7YowhInIJHQCtvB0KlcgK+AGdlC6DiIjKgzGGiMhVdAS8io4oK1x/2TX3xyzWmPICGgOB
ytRCRET3R2W1uubNi4jIFd0E9gCpNqtyFa5g5mpfa5mBAkADqAFvQA+EAY8pXRUREZWTq922iIhc
Wk0gTN4aRRpdVriTjOh6PRlpV1AtYCxAADMMEZFDYYwhInItbYHagBZwB7TyzphmpauqfipAB+gB
T6CeO7oqXQ8REd0XDiojInJF+4FbQC6QB5gA0WYRMxf5fssKGK2oq4IvmGGIiByPi9ytiIioiGeA
YEDvwnvIuAOBKrgDXZSuhIiIKoAxhojIRbUHNPIqzFabhZgLZ8s4WbNeuOvHE+jIGyERkWPioDIi
Ipe2G7gCGAANAMBiM+NfcKK9MgvfiBrQAQC8gacBX8UqIiKiB8IYQ0Tk0qzAHiANyAVMgEruTjjZ
vUGKZNIb9AJ8gY6An9JVERFRhbGXTkTk0lTA3wBfKzSAG6CWR5ppnasbA/ntqIB8Ed2YYYiIHBy7
MUREBABHgT9EqAGrAAtgAcxONPVfAAQzfNwQyDn9REROgTGGiIj+kgQkAVlAjrzDvS3bZQDszT3X
WxOAEKA+0KJ6CiIioirGGENERP+TDsQBGUA24Aa4yc9Le8toyvpVxdw98q0wcVnlXT7dgTZAaHWX
RkREVYUxhoiIijsCJAO3ADdAA5iBPACAVuG6SibYPJBWWlMDZkAFaIBAIBh4QskCiYio8jHGEBFR
CVKA48AtIB8wAAAEe40xAKyAKK9aI22DYwY8gAAgDHhU4eqIiKjyMcYQEVGpDgF/AtmAVt5Jpjvy
U7wAABLESURBVBh7WANAtIkx0vgxFWAB6gKPAUFKl0dERFWBMYaIiMpyGYgHUkRYhOIxxmpPMQby
UtHegDvQHGigcF1ERFSFGGOIiOjeTgEJQEHRJ8uOMVLmsdz/TpqFy45JDZa7l0crcV0yAfACvIEQ
oM19XpGIiBwOYwwREZXXH8AleTnmfJsgIWUVC+AJqAGL/JIIWIqeoexbjpRP1PI5jTaXcJP3r5SC
jRXIA0R5p04zUANoDLSslPdJRER2jzGGiIjuz3XgN+AOYJEXYhbl6OIJCDYZxlTKGUq78djGGAD5
cpJRy6s/6wABUAGivBaZNGMnnB0YIiIXwxhDREQVYQZOApcBE2CWk4zUqJHSiNQ/cSvpd8uOMYUz
cKSMZATMgFVeKk0LCIAO8AGCgKaAbyW/MyIicgCMMURE9EBuA78DN4A7gEkeYCalGinGFA4SkxSm
lBJvP4J8gMrmQYF8Zk/AG/AEWgE1q/A9ERGRvWOMISKiynEVSAaygHwgFSiQR3xJP2Y5t0hT9oXS
ezJaOf94AABMgNYKdxU8gUbAQ9XxVoiIyN4xxhARUeXLAU4DN4EbgFrej1IiyPGmtG6MGtAAGiAY
qAnUAgKrr3AiInIMjDFERFRNCoBU4DpglQeJqeQp+wKgBwKAALkJQ0REVAbGGCIiIiIicjDqex9C
RERERERkTxhjiIiIiIjIwTDGEBERERGRg2GMISIiIiIiB8MYQ0REREREDoYxhoiIiIiIHAxjDBER
ERERORjGGCIiIiIicjCMMURERERE5GAYY4iIiIiIyMEwxhARERERkYNhjCEiIiIiIgfDGENERERE
RA6GMYaIiIiIiBwMYwwRERERETkYxhgiIiIiInIwjDFERERERORgGGOIiIiIiMjBMMYQEREREZGD
YYwhIiIiIiIHwxhDREREREQOhjGGiIiIiIgcDGMMERERERE5GMYYIiIiIiJyMIwxRERERETkYBhj
iIiIiIjIwTDGEBERERGRg2GMISIiIiIiB8MYQ0REREREDsZN6QKIiKi6derU6fr162az2fbJdevW
PfPMMw94ZovFcvXq1bS0tPr16wcGBj7g2UqUnp6emJjo4+PTokWLu1997rnnEhISbJ/RaDTr16/v
0KEDgISEhBs3blhF0apWWy0WlbrId3lBNWq0bt26KmomIqJKxxhDRORykpOTr1y54uHhIf2nIAgA
iqUa24MHDRoEoHbt2lFRUaWdMy4u7vPPPz927FhGRobRaAwODvb19X322Wdnz57t6+v74DWbTKaI
iIg9e/YkJydbLBaDwRAUFNSlS5eJEyeGh4cXHpaSkpKdnZ2TkwPAw8PDYDAAyM3NlV4dNGjQ77//
DsDLyysnJ8fLy0sURaPRKL1av379P//888FLJSKiasAYQ0TkigRBMBgMHh4eQUFBarUagJeXV4lH
Tp069dSpUwCysrJKO9uMGTOmT58uZQYAWq3WZDJdvnx5w4YNu3btWrZsWffu3R+k2ps3bz7//PNn
zpwxGAxarVYQBFEU09LSfvnll507d06bNm3cuHHSkcHBwbm5uf7+/pmZmQaDQYorhecJCwurU6dO
ZmamSq0+/fvvarVao9HUrVs3ODgYQJ06dR6kSCIiqk4qq9WqdA1ERFStGjZseO3aNW9v7wYNGvz3
v/8t48jIyMg333xTatfUr1+/2HgtydmzZ9u1a+fu7p6enh4YGBgYGOjn55eZmZmRkZGXlyeKYlhY
2IkTJzQaTYULDgsLu3DhAgBPT89GjRrVqlUrPSMj/rffpFRTs2bNmJiYRo0a2f7KJ598Mnv2bABG
o3H79u09e/Ysds4nOnQ4euRInTp1rl69WuHCiIhIKZziT0TkctRqtdFovH37dhnDvU6ePDl8+PCJ
EydKfRuDwVDaqLORI0fm5OSoVKrQ0NDVq1f/8ccfR48eTUpK+nv//hqNRqvVnj9//m9/+1uFq501
a9bVq1dFUdRqtWPHjj1x4sS2bduOHD68fPnywMBAo9F4586diIiIYr8VFhYmiqIUwIrNgbGl1+sr
XBgRESmIg8qIiFyOoFKV8WpAQEBmZqY0cAulDzaTREVFxcXFBQYG3rp167PPPnvxxRcLX/p6yZJD
cXGJiYmCIBw7dqzC1f7000+CIHh5edWtW/ff//534fOvv/76yZMnV65cmZube/jwYZPJVFrDR13m
+yUiIkfEbgwRERUhCEJgYKDUxygMM6WJiYkBcPv2bQ8Pj6lTpxZ7tU2bNlqtFoBGo7l582bF6klJ
STEYDDk5Of379y/20ogRI6SGUkpKytKlS21fUqvVgo2KXZqIiOwWuzFERFRE48aNLVZrQX6+IAjZ
2dlJSUllxICkpCRRFD08PLy9ve9+NScnRxrZZTKZKjYVMyYmJiMjQ6vVGo3GXr16FXu1SZMmNWrU
uHXrlsFgOHnyZIln0Gq1mpLqt1osACwWSwWqIiIixTHGEBFREYcOHbL9z9DQUKmRoippaNaIESO6
deum1WpLnDmTmJgIQOrnhISEVKCYgwcPGo1GQRB8fHzatWt39wFBQUGXLl0yGAzXr1+3fd5sNksr
KYuiaCqzoURERI6IMYaIiMpisVikboy6pInyAwcOLO0X165de+3aNUEQcnJy7l4orJyys7Olq+t0
utKOyc/PFwQhPT3d9kl3d3cAgiAUbgtDRETOhHNjiIioLBUbdrV27drx48fn5eWp1eqwsLDt27dX
7Oomk0l6UGKIAlA49aVYjHFzcwOg0+k4MYaIyCkxxhARUVmsVqtGo9FoNCUOKivR+PHjhw4dmp+f
DyAwMHDZsmUPUoAoiqIolpamLBaL1G+RckuJv/sgVyciIvvEQWVERFRp/nv8+BtDh16+fNnDw0On
04WEhBw8eDAgIKDCJ/T29pYWCShthYDCOTDSkmglKvF3PT09vby8GHKIiBwUuzFERFQWs9msKt+4
smnTpj3bpcvly5dzcnJ8fX1ffPHFM2fOPEiGAVC47nNBQUGJB0ijzrRabbFAIkUXS+nLkWVlZUkZ
6UHKIyIipbAbQ0RE92AURY1aXeKorUJ9+/bdvHmzh4eHwWAIDw9ftmxZiQuL3a82bdr4+PiYTKas
rKz4+PiWLVsWOyA9PV2r1Wq12rCwMNvnpSFw0oyaEufVHD9+vKCgQFoJgIiIHA67MUREdA9Sy6KM
uTFPPvnkkSNHvLy8jEZjhw4dTp48WSkZBkCXLl0sFovBYPD09Fy7dm2xV1NSUpKTk41Go8FgaNiw
oe1L5dmmhhmGiMhxMcYQEdE9SDGmtLXCBg8efOzYsdzcXL1e36dPn2LbzjwgvV7frFkzf39/i8Wy
adOmYq/OmDEjKysLQEhIyOTJk21f0mq1giBotVoPDw9tmX0kIiJyRIwxRER0D9IElRJnmOzfv3/z
5s2iKGZlZbm7u7/55pv3PNvEiRPbt2/fvn37Jzp02Lx58z2P/9vf/pabmyuK4p07d2bMmFH4fHx8
/A8//AAgODi4ffv2QUFBtr9lNBpFUZQaNcaStuYkIiKHxi+oiIjoHqQYU+IwrenTp0PevCUzM3PA
gAEajQZy5lGr1SqVShTF/Pz83bt3d+rUCUBSUtKvv/6q1Wo1Gk12dvY9rz59+vT169dfv349MzNz
7ty5R48ebdGixcWLF3fs2JGZmanVak0m08cff1y5b5mIiOwcYwwREd2DIAgWi8VcUk8jMTGxcLEv
aeFjabsY6Ukp3kiNkfy8POkwjUbj4eEhNXDyS1l/rJgtW7a88soriYmJ2dnZO3fu3Llzp9Fo9PDw
ABASEvLee++1atWq2K9IM3mkAsq/4w0RETkKDiojIqJ7UKvVJU6MSU9Pz8vLk6JC+c+WnZ2dk5Nj
MBjK/yuPPvrorl27OnbsKCUinU4nXTE0NHTWrFnjx48v/6mIiMg5sBtDRERlSUtLK+2lwMDAjIyM
Yk9aLBa1Wm2RWjdqtVqtNoui2WTS6XTSATt37rxy5crw4cOPHDlSMzi4nGWEhoYePHgwJSUlNjbW
aDRqtdoWLVo0bdq0tOOlnk9WVtb9piwiInIIjDFERK5I2mulxHFiD+ivrVpsFgdzEwS3okEiNDQ0
Nzc3ICDg2Wefva+Th4SE9O/fvzxHGo1GqXUjimJVvE0iIlIWYwwRketRq729vdPT02/evLlgwQJp
ja9BgwZV2/VHjhx5+PDhV1991dvbu3LPvH///qtXrxqNxmXLlomi6OHhYTQaOTeGiMj5qMqzQRgR
ETmTpk2bXrhwodiTR44cuXuifFWYMGHC119/3aNHj7v3gXlwbdu2vXjxYl5eniiK0lgyo9G4a9eu
rl27Vvq1iIhIQezGEBG5nPy8PKPR6OPjU+TJ/PzquXqtWrWWL1/++uuvV8XJBTc3jUaj0+mysrKk
1dL8/f05N4aIyPmwG0NE5HK+++677OzsYouPvf3220rVU4k2b96cmppqMpmsVmvhmsujRo1Sui4i
IqpkjDFERERERORguG8MERERERE5GMYYIiIiIiJyMIwxRERERETkYBhjiIiIiIjIwTDGEBERERGR
g2GMISIiIiIiB8MYQ0REREREDoYxhoiIiIiIHAxjDBERERERORjGGCIiIiIicjCMMURERERE5GAY
Y4j+v737B41qy+MAfid35g6zmuCfIrBiiIWJfxH1PdAIqaIWgiKx3VdERiSmEW1UAhYS8FmIb0Gw
SWMjIYH0CoqpYxARC0FkJGIigYgZMjvZq1sMK8F9Pjbjy7ve5POpTs495PebavhyzpkLAEDKiDEA
AEDKiDEAAEDKZJNuAACWYGhoqFKpdHZ2tra2Jt0LAImxGwNAaty4ceP06dPFYvH6r78m3QsASRJj
AEiNwcHBarVarVaTbgSAhIkxAKTAxPh4W1vb1NRULcP8e2Eh6Y4ASJK7MQD8uIaHh1+/fj06Ovrk
yZP5+fmmpqYoisIwTLovABImxgDwg1pYWDh37tzs7Gwcx0EQhGFYLpfjOI6i6NOnT0l3B0CSHCoD
YMkOHz7c3Nzc0tKyb9++/306MTHx9/8aGBiou0qmoWFhYeF3b8I0NPj+AljV7MYAsGQHDx58/Phx
EARv3rx58OBBV1fX4qfDw8MfP36M4ziXy124cKHuKtkwHB0drVar5XK5Vuvq1aszMzOu+AMgxgCw
ZD09PdevX6+Nh4aGvooxDx8+nJubKxQKnZ2d+Xz+ewp1dnYu/vPOnTtfzpgBsJrZlAdgyVpbW7dt
21a7pjI2Nrb40du3b58/f75+/frGxsZisVibvHTpUjabzWaz+Xz+b9+QyWQymUxHR8f/00Amk/nz
PxUA6SHGAFCPc319YRhGUfThw4dnz559mb99+3a5XK5UKhs3bjx+/HhtMpfLRVFU+5Gxb6ktWLNm
TUIfCIA0cagMgHqcKRb/+dtvpVKpXC4PDg7evHmzNj84OBhFURzHi0+a7dmz59ixY7Xxt27nh2EY
x/GuXbuWu3MAVgAxBoA6HTp0qFQqBUFw//792szU1NT09HShUMjlcv/45ZcvK7u7u7u7u5PpEoCV
yKEyAOp06tSp2itcXr58OTExEQTB5cuX4ziuVqvNzc0///RT0g0CsGKJMQDUqaurq6GhIZfLNTU1
XblyJQiCp0+fFgqFMAzb29uT7g6AlcyhMgDqd+TIkaGhoWq1OjY2duvWrXfv3tXm7969u3jZyMjI
vXv3auM/vhuzdevWa9euLWvPAKwAYgwA9evv7x8ZGVm7du38/PzAwMDs7GwURQcOHFi3bt3iZY8e
PRoeHv7jf1UoFObn5796BQ0A/C4xBoD6tbW17dixY3x8PAzDmZmZIAjy+Xxvb+9Xy9rb23fu3Fkb
f+uVL7lcLo7jLVu2LGvDAKwMYgwA36W3t/fMmTO1cRiGGzZsOHny5Fdr+vr6+vr6/vLWAFixXPEH
4Lv09PRs2rSpNo7jeO/evctXK5vNRlFUK7R8VQD48dmNAeB7bd68eXJyMgiCOI6LxeLyFTpx4sTU
9PS/KpWjR48uXxUAfnyZz58/J90DAOm2e/fuFy9eBEEQx7GvFQD+Ag6VAfBdzp8/XyqVaj+XfPHi
xaTbAWBVsBsDQD06OjomJyer1er79+9rGaalpeXVq1dJ9wXAqmA3BoB6VCqVUqk0NzcXx3EYhkEQ
9Pf3J90UAKuFK/4A1GP79u1RPt+QyTQ2Nra0tJw9e3b//v1JNwXAauFQGQAAkDIOlQEAACkjxgAA
ACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkj
xgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAAACkjxgAA
ACkjxgAAACkjxgAAACkjxgAAACnzH++pRi8nI0ntAAAAAElFTkSuQmCC'/>


In [4]:
# four nodes in the pic above
# Each carries a two-dimensional feature vector and a label y, representing the class to which this node belongs

# first define node
x=torch.tensor([[2,1],[5,6],[3,7],[12,0]],dtype=torch.float64)
y=torch.tensor([0,1,0,1],dtype=torch.float64)
# then define edge

# first way to create edge
# first dimension is the source node of the edge
# second dimension is the target node of the edge
edge_index1=torch.tensor([[0,1,2,0,3],[1,0,1,3,2]],dtype=torch.long)
data1=Data(x=x,y=y,edge_index=edge_index1)
# second way to create edge
# each dimension contains source node and target node
# need to use contiguous() method to ensure that the input tensor is stored continuously in memory
edge_index2=torch.tensor([[0,1],[1,0],[2,1],[0,3],[2,3]],dtype=torch.long)
data2=Data(x=x,y=y,edge_index=edge_index2.contiguous())


In [6]:
# try to do the output
# output are all the dimension of each graph data
print(data1)
print(data2)

Data(x=[4, 2], edge_index=[2, 5], y=[4])
Data(x=[4, 2], edge_index=[5, 2], y=[4])


dataset, use ENZYMES dataset as an example

In [8]:
enDataset=TUDataset(root='.\ENZYMES',name='ENZYMES')

print(enDataset)
print(len(enDataset))
print(enDataset.num_classes)
print(enDataset.num_node_features)

Extracting ENZYMES\ENZYMES\ENZYMES.zip
Processing...


ENZYMES(600)
600
6
3


Done!


check structure of one graph

In [9]:
dataCheck=enDataset[14]
print(dataCheck)
print(dataCheck.is_undirected())

Data(edge_index=[2, 128], x=[36, 3], y=[1])
True


Include 36 nodes, each node has 3 features, total 128/2=64 edges (undirected graph), graph is categorized as Category 1

2 way to make dataset shuffle

In [10]:
# first way just call shuffle
dShuffle1=enDataset.shuffle()
# second way use torch.randperm
perm=torch.randperm(len(enDataset))
dShuffle2=enDataset[perm]


Batches

In [19]:
enDataset=TUDataset(root='./ENZYMES/',name='ENZYMES',use_node_attr=True)
enDataloader=torch_geometric.loader.DataLoader(enDataset,batch_size=128,shuffle=True)
# UserWarning: 'data.DataLoader' is deprecated, use 'loader.DataLoader' instead
# enDataloader=torch_geometric.data.DataLoader(enDataset,batch_size=128,shuffle=True)

for enData in enDataloader:
    print(enData)
    print(enData.num_graphs)
    x=scatter_mean(enData.x,enData.batch,dim=0)
    print(x.size())

DataBatch(edge_index=[2, 16342], x=[4297, 21], y=[128], batch=[4297], ptr=[129])
128
torch.Size([128, 21])
DataBatch(edge_index=[2, 16282], x=[4343, 21], y=[128], batch=[4343], ptr=[129])
128
torch.Size([128, 21])
DataBatch(edge_index=[2, 14902], x=[3899, 21], y=[128], batch=[3899], ptr=[129])
128
torch.Size([128, 21])
DataBatch(edge_index=[2, 15602], x=[4043, 21], y=[128], batch=[4043], ptr=[129])
128
torch.Size([128, 21])
DataBatch(edge_index=[2, 11436], x=[2998, 21], y=[88], batch=[2998], ptr=[89])
88
torch.Size([88, 21])


first batch contains 4297 nodes, each node has 21 features, total 128 graphs

First GNN with cora dataset

In [24]:
# dataset preparation
CoraDataset=Planetoid(root='./Cora',name='Cora')
print(CoraDataset)
print(len(CoraDataset))

Cora()
1


In [33]:
class myGNN(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.GCN1=torch_geometric.nn.GCNConv(CoraDataset.num_node_features,16)
        self.GCN2=torch_geometric.nn.GCNConv(16,CoraDataset.num_classes)
        self.dropout=torch.nn.Dropout(0.4)
        self.relu=torch.nn.ReLU()
        self.training=0.1
        self.logsoftmax=torch.nn.LogSoftmax(dim=1)
    def forward(self,data):
        x,edge_index=data.x,data.edge_index
        x=self.GCN1(x,edge_index)
        x=self.relu(x)
        x=self.dropout(x)
        # x=torch.nn.Dropout(x,training=self.training)
        
        x=self.GCN2(x,edge_index)
        return self.logsoftmax(x)

In [39]:
# In cora dataset,train_mask denotes training set, val_mask denotes validation set and test_mask denotes test set.

device=torch.device('cuda' if torch.cuda.is_available() else 'cpu')
GNNmodel=myGNN().to(device)
CoraData=CoraDataset[0].to(device)
optimizer=torch.optim.Adam(GNNmodel.parameters(),lr=0.01,weight_decay=5e-4)
loss_function=torch.nn.NLLLoss()
GNNmodel.train()
for epoch in range(200):
    
    # CoraData=CoraDataset.to(device)
    optimizer.zero_grad()
    pred=GNNmodel(CoraData)
    loss=loss_function(pred[CoraData.train_mask],CoraData.y[CoraData.train_mask])
    # loss=torch.nn.NLLLoss(pred[CoraData.train_mask],CoraData.y[CoraData.train_mask])
    loss.backward()
    optimizer.step()
GNNmodel.eval()
_,p=GNNmodel(CoraData).max(dim=1)
# print(p)
correct=int(p[CoraData.test_mask].eq(CoraData.y[CoraData.test_mask]).sum().item())
acc=correct/int(CoraData.test_mask.sum())
print('Accuracy:{:.4f}'.format(acc))

Accuracy:0.7940


GCN model(message passing)

In [ ]:
class myGCN(MessagePassing):
    def __init__(self,in_channels,out_channels):
        # Specify the aggregation function as additive(Standard implementation of Graph Convolutional Networks)
        super().__init__(aggr='add')
        self.linear=torch.nn.Linear(in_channels,out_channels)
    def forward(self,x,edge_index):
        # Preprocessing of the adjacency matrix. Add an edge from all nodes to itself
        edge_index,_=add_self_loops(edge_index,num_nodes=x.siza(0))
        x=self.linear(x)
        # The following lines of code normalize the transformed node features
        row,col=edge_index
        # Calculate the degree of each node
        deg=degree(col,x.size(0),dtype=x.dtype)
        # Calculating the inverse of the square root of a degree
        deg_inv_sqrt=deg.pow(-0.5)
        # Calculate the normalization factor for each edge
        norm=deg_inv_sqrt[row]*deg_inv_sqrt[col]
        # use propagate method to pass the message
        return self.propagate(edge_index,x=x,norm=norm)
    def message(self,x_j,norm):
        # Calculate the information coming from neighboring nodes
        return norm.view(-1,1)*x_j

edge convolution

In [ ]:
class myEdgeConv(MessagePassing):
    def __init__(self, in_channels, out_channels):
        # For each node, its new feature is determined by the one with the largest difference among its neighboring nodes. This is the standard architecture of EdgeConv
        super().__init__(aggr='max')
        self.mlp = torch.nn.Sequential(
            # The features of both the source and target nodes are taken into account when calculating the difference, so the dimension of the input features is twice the dimension of the original features
            torch.nn.Linear(2*in_channels, out_channels),
            torch.nn.ReLU(),
            #  Keeping the number of input and output channels the same can be viewed as an operation that linearly transforms features in a high-dimensional feature space.
            torch.nn.Linear(out_channels, out_channels))

    def forward(self, x, edge_index):
        return self.propagate(edge_index, x=x)

    def message(self, x_i, x_j):
        # Calculate the difference between the source and target node features and stitch them together with the source node features.
        # Differences between source and target node features are computed to capture local relationships between nodes.
        # The new features of each node then depend not only on the features of its neighboring nodes, but also on the interrelationships between the neighboring nodes.
        # This approach better captures the local structural information in the graph and thus improves the performance of the model.
        tmp = torch.cat([x_i, x_j-x_i], dim=1)
        # Transformation of spliced features using multilayer perceptron
        return self.mlp(tmp)


dynamic edge convolution

In [ ]:
class myDynamicEdgeConv(myEdgeConv):
    def __init__(self,in_channels,out_channels,k=6):
        super().__init__()
        self.k=k
    def forward(self,x,batch=None):
        # In order to dynamically construct the neighbor relationship of each node, so as to efficiently capture the relationship between nodes in dynamic graph convolution and achieve more flexible graph convolution operations
        edge_index=knn_graph(x,self.k,batch,loop=False,flow=self.flow)
        return super().forward(x,edge_index)